---
title: The LOTUS Initiative for Open Knowledge Management in Natural Products Research
keywords:
- Open Science
- Natural Products
- Wikidata
- Knowledge Graph
- Linked Data
- LOTUS Initiative
lang: en-US
date-meta: '2022-04-21'
author-meta:
- Adriano Rutz
- Maria Sorokina
- Jakub Galgonek
- Daniel Mietchen
- Egon Willighagen
- Arnaud Gaudry
- James G. Graham
- Ralf Stephan
- Roderic Page
- Jiří Vondrášek
- Christoph Steinbeck
- Guido F. Pauli
- Jean-Luc Wolfender
- Jonathan Bisson
- Pierre-Marie Allard
header-includes: |-
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="The LOTUS Initiative for Open Knowledge Management in Natural Products Research" />
  <meta name="citation_title" content="The LOTUS Initiative for Open Knowledge Management in Natural Products Research" />
  <meta property="og:title" content="The LOTUS Initiative for Open Knowledge Management in Natural Products Research" />
  <meta property="twitter:title" content="The LOTUS Initiative for Open Knowledge Management in Natural Products Research" />
  <meta name="dc.date" content="2022-04-21" />
  <meta name="citation_publication_date" content="2022-04-21" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Adriano Rutz" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0003-0443-9902" />
  <meta name="twitter:creator" content="@adafede" />
  <meta name="citation_author" content="Maria Sorokina" />
  <meta name="citation_author_institution" content="Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany" />
  <meta name="citation_author_orcid" content="0000-0001-9359-7149" />
  <meta name="twitter:creator" content="@ms_sorok" />
  <meta name="citation_author" content="Jakub Galgonek" />
  <meta name="citation_author_institution" content="Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic" />
  <meta name="citation_author_orcid" content="0000-0002-7038-544X" />
  <meta name="twitter:creator" content="@JGalgonek" />
  <meta name="citation_author" content="Daniel Mietchen" />
  <meta name="citation_author_institution" content="Ronin Institute, 127 Haddon Pl, Montclair, NJ 07043, United States" />
  <meta name="citation_author_institution" content="Leibniz Institute of Freshwater Ecology and Inland Fisheries, Müggelseedamm 310, D-12587 Berlin" />
  <meta name="citation_author_institution" content="School of Data Science, University of Virginia, Dell 1 Building, Charlottesville, VA 22904, United States" />
  <meta name="citation_author_orcid" content="0000-0001-9488-1870" />
  <meta name="twitter:creator" content="@EvoMRI" />
  <meta name="citation_author" content="Egon Willighagen" />
  <meta name="citation_author_institution" content="Dept of Bioinformatics-BiGCaT, NUTRIM, Maastricht University, Universiteitssingel 50, NL-6229 ER, Maastricht, The Netherlands" />
  <meta name="citation_author_orcid" content="0000-0001-7542-0286" />
  <meta name="twitter:creator" content="@egonwillighagen" />
  <meta name="citation_author" content="Arnaud Gaudry" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-3648-7362" />
  <meta name="twitter:creator" content="@gaudarn" />
  <meta name="citation_author" content="James G. Graham" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0002-7114-8921" />
  <meta name="citation_author" content="Ralf Stephan" />
  <meta name="citation_author_institution" content="Ontario Institute for Cancer Research (OICR), 661 University Ave Suite 510, Toronto, Canada" />
  <meta name="citation_author_orcid" content="0000-0002-4650-631X" />
  <meta name="citation_author" content="Roderic Page" />
  <meta name="citation_author_institution" content="IBAHCM, MVLS, University of Glasgow, Glasgow, United Kingdom" />
  <meta name="citation_author_orcid" content="0000-0002-7101-9767" />
  <meta name="twitter:creator" content="@rdmpage" />
  <meta name="citation_author" content="Jiří Vondrášek" />
  <meta name="citation_author_institution" content="Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic" />
  <meta name="citation_author_orcid" content="0000-0002-6066-973X" />
  <meta name="citation_author" content="Christoph Steinbeck" />
  <meta name="citation_author_institution" content="Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany" />
  <meta name="citation_author_orcid" content="0000-0001-6966-0814" />
  <meta name="twitter:creator" content="@csteinbeck" />
  <meta name="citation_author" content="Guido F. Pauli" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0003-1022-4326" />
  <meta name="citation_author" content="Jean-Luc Wolfender" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-0125-952X" />
  <meta name="citation_author" content="Jonathan Bisson" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0003-1640-9989" />
  <meta name="twitter:creator" content="@Bjonnh" />
  <meta name="citation_author" content="Pierre-Marie Allard" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Department of Biology, University of Fribourg, Chemin du Musée 10, 1700 Fribourg, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0003-3389-2191" />
  <meta name="twitter:creator" content="@allard_pm" />
  <link rel="canonical" href="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta property="og:url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta property="twitter:url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta name="citation_pdf_url" content="https://lotusnprod.github.io/lotus-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://lotusnprod.github.io/lotus-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://lotusnprod.github.io/lotus-manuscript/v/f70d777079a4cb780ba736d2a345f16bda970e0e/" />
  <meta name="manubot_html_url_versioned" content="https://lotusnprod.github.io/lotus-manuscript/v/f70d777079a4cb780ba736d2a345f16bda970e0e/" />
  <meta name="manubot_pdf_url_versioned" content="https://lotusnprod.github.io/lotus-manuscript/v/f70d777079a4cb780ba736d2a345f16bda970e0e/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <meta property="og:image" content="https://github.com/lotusnprod/lotus-manuscript/raw/f70d777079a4cb780ba736d2a345f16bda970e0e/content/images/thumbnail.png" />
  <meta property="twitter:image" content="https://github.com/lotusnprod/lotus-manuscript/raw/f70d777079a4cb780ba736d2a345f16bda970e0e/content/images/thumbnail.png" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
bibliography:
- content/manual-references.json
manubot-output-bibliography: output/references.json
manubot-output-citekeys: output/citations.tsv
manubot-requests-cache-path: ci/cache/requests-cache
manubot-clear-requests-cache: false
...






<small><em>
This manuscript
([permalink](https://lotusnprod.github.io/lotus-manuscript/v/f70d777079a4cb780ba736d2a345f16bda970e0e/))
was automatically generated
from [lotusnprod/lotus-manuscript@f70d777](https://github.com/lotusnprod/lotus-manuscript/tree/f70d777079a4cb780ba736d2a345f16bda970e0e)
on April 21, 2022.
</em></small>

## Authors



+ **Adriano Rutz** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-0443-9902](https://orcid.org/0000-0003-0443-9902)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [adafede](https://github.com/adafede)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [adafede](https://twitter.com/adafede)
    \
  <small>
     School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
     · Funded by CRSII5_189921
  </small>

+ **Maria Sorokina** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-9359-7149](https://orcid.org/0000-0001-9359-7149)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [mSorok](https://github.com/mSorok)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [ms_sorok](https://twitter.com/ms_sorok)
    \
  <small>
     Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany
     · Funded by CRC1127
  </small>

+ **Jakub Galgonek** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-7038-544X](https://orcid.org/0000-0002-7038-544X)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [galgonek](https://github.com/galgonek)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [JGalgonek](https://twitter.com/JGalgonek)
    \
  <small>
     Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic
     · Funded by LM2018131
  </small>

+ **Daniel Mietchen** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-9488-1870](https://orcid.org/0000-0001-9488-1870)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [Daniel-Mietchen](https://github.com/Daniel-Mietchen)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [EvoMRI](https://twitter.com/EvoMRI)
    \
  <small>
     Ronin Institute, 127 Haddon Pl, Montclair, NJ 07043, United States; Leibniz Institute of Freshwater Ecology and Inland Fisheries, Müggelseedamm 310, D-12587 Berlin; School of Data Science, University of Virginia, Dell 1 Building, Charlottesville, VA 22904, United States
     · Funded by G-2019-11458
  </small>

+ **Egon Willighagen** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-7542-0286](https://orcid.org/0000-0001-7542-0286)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [egonw](https://github.com/egonw)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [egonwillighagen](https://twitter.com/egonwillighagen)
    \
  <small>
     Dept of Bioinformatics-BiGCaT, NUTRIM, Maastricht University, Universiteitssingel 50, NL-6229 ER, Maastricht, The Netherlands
     · Funded by G-2019-11458
  </small>

+ **Arnaud Gaudry** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-3648-7362](https://orcid.org/0000-0002-3648-7362)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [ArnaudGaudry](https://github.com/ArnaudGaudry)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [gaudarn](https://twitter.com/gaudarn)
    \
  <small>
     School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
     · Funded by CRSII5_189921
  </small>

+ **James G. Graham** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-7114-8921](https://orcid.org/0000-0002-7114-8921)
      \
  <small>
     Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
     · Funded by U41 AT008706 and AT000155
  </small>

+ **Ralf Stephan** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-4650-631X](https://orcid.org/0000-0002-4650-631X)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [rwst](https://github.com/rwst)
      \
  <small>
     Ontario Institute for Cancer Research (OICR), 661 University Ave Suite 510, Toronto, Canada
  </small>

+ **Roderic Page** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-7101-9767](https://orcid.org/0000-0002-7101-9767)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [rdmpage](https://github.com/rdmpage)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [rdmpage](https://twitter.com/rdmpage)
    \
  <small>
     IBAHCM, MVLS, University of Glasgow, Glasgow, United Kingdom
  </small>

+ **Jiří Vondrášek** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-6066-973X](https://orcid.org/0000-0002-6066-973X)
      \
  <small>
     Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic
     · Funded by LM2018131
  </small>

+ **Christoph Steinbeck** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-6966-0814](https://orcid.org/0000-0001-6966-0814)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [steinbeck](https://github.com/steinbeck)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [csteinbeck](https://twitter.com/csteinbeck)
    \
  <small>
     Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany
     · Funded by CRC1127
  </small>

+ **Guido F. Pauli** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-1022-4326](https://orcid.org/0000-0003-1022-4326)
      \
  <small>
     Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
     · Funded by U41 AT008706 and AT000155
  </small>

+ **Jean-Luc Wolfender** \
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-0125-952X](https://orcid.org/0000-0002-0125-952X)
      \
  <small>
     School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
     · Funded by CRSII5_189921
  </small>

+ **Jonathan Bisson** ^[✉](#correspondence)^\
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-1640-9989](https://orcid.org/0000-0003-1640-9989)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [bjonnh](https://github.com/bjonnh)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [Bjonnh](https://twitter.com/Bjonnh)
    \
  <small>
     Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
     · Funded by U41 AT008706 and AT000155
  </small>

+ **Pierre-Marie Allard** ^[✉](#correspondence)^\
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-3389-2191](https://orcid.org/0000-0003-3389-2191)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [oolonek](https://github.com/oolonek)
    · ![Twitter icon](images/twitter.svg){.inline_icon width=16 height=16}
    [allard_pm](https://twitter.com/allard_pm)
    \
  <small>
     School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Department of Biology, University of Fribourg, Chemin du Musée 10, 1700 Fribourg, Switzerland
     · Funded by CRSII5_189921
  </small>


::: {#correspondence}
✉ — technical correspondence preferred *via* [GitHub Issues](https://github.com/lotusnprod/lotus-processor/issues).
Otherwise, address correspondence to <research@bjonnh.net>, and <pierre-marie.allard@unifr.ch>.
:::


## Abstract {.page_break_before}

Contemporary bioinformatic and chemoinformatic capabilities hold promise to reshape knowledge management, analysis and interpretation of data in natural products research.
Currently, reliance on a disparate set of non-standardized, insular, and specialized databases presents a series of challenges for data access, both within the discipline and for integration and interoperability between related fields.
The fundamental elements of exchange are referenced structure-organism pairs that establish relationships between distinct molecular structures and the living organisms from which they were identified.
Consolidating and sharing such information *via* an open platform has strong transformative potential for natural products research and beyond.
This is the ultimate goal of the newly established LOTUS initiative, which has now completed the first steps toward the harmonization, curation, validation and open dissemination of 750,000+ referenced structure-organism pairs.
LOTUS data is hosted on Wikidata and regularly mirrored on [https://lotus.naturalproducts.net](https://lotus.naturalproducts.net).
Data sharing within the Wikidata framework broadens data access and interoperability, opening new possibilities for community curation and evolving publication models.
Furthermore, embedding LOTUS data into the vast Wikidata knowledge graph will facilitate new biological and chemical insights.
The LOTUS initiative represents an important advancement in the design and deployment of a comprehensive and collaborative natural products knowledge base.


## Introduction {.page_break_before}

### Evolution of Electronic Natural Products Resources

Natural Products (NP) research is a transdisciplinary field with wide-ranging interests: from fundamental structural aspects of naturally-occurring molecular entities to their effects on living organisms and extending to the study of chemically-mediated interactions within entire ecosystems.
Defining the "natural" qualifier is a complex task [@doi:10.1057/s41599-020-0390-y; @doi:10/bkfd8h].
We thus adopt here a broad definition of a NP as any chemical entity *found in* a living organism, hereafter refered to as a _structure-organism pair_.
An additional and fundamental element of a structure-organism pair is a reference to the experimental evidence that establishes the linkages between the chemical structure and the biological organism.
A future-oriented electronic NP resource should contain fully-referenced structure-organism pairs.

Reliance on data from the NP literature presents many challenges.
The assembly and integration of NP occurrences into an inter-operative platform relies primarily on access to a heterogeneous set of databases (DB) whose content and maintenance status are critical factors in this dependency [@doi:10/gfwm27].
A tertiary inter-operative NP platform is thus dependent on a secondary set of data that has been selectively annotated into a DB from primary literature sources.
The experimental data itself reflects a complex process involving collection or sourcing of natural material (and establishment of its identity), a series of material transformation and separation steps and ultimately the chemical or spectral elucidation of isolates.
The specter of human error and the potential for the introduction of biases are present at every phase of this journey.
These include publication biases [@doi:10.1002/asi.22784], such as emphasis on novel and/or bioactive structures in the review process, or, in DB assembly stages, with selective focus on a specific compound class or a given taxonomic range, or disregard for annotation of other relevant evidence that may have been presented in primary sources.
Temporal biases also exist: a technological "state-of-the-art" when published can eventually be recast as anachronistic.

The advancement of NP research has always relied on the development of new technologies.
In the past century alone, the rate at which unambiguous identification of new NP entities from biological matrices can be achieved has been reduced from years to days and in the past few decades, the scale at which new NP discoveries are being reported has increased exponentially.
Without a means to access and process these disparate NP data points, information is fragmented and scientific progress is impaired [@doi:10.1371/journal.pone.0118747].
To this extent, contemporary bioinformatic tools enable the (re-)interpretation and (re-)annotation of (existing) datasets documenting molecular aspects of biodiversity [@doi:10.1038/s41598-021-87796-6; @doi:10.1038/s41592-020-0916-7].

While large, well-structured and freely accessible DB exist, they are often concerned primarily with chemical structures (e.g. [PubChem](https://pubchem.ncbi.nlm.nih.gov) [@doi:10.1093/nar/gky1033], with over 100M entries) or biological organisms (e.g. [GBIF](https://www.gbif.org) [@GBIFOrg2020], with over 1,900M entries), but scarce interlinkages limit their application for documentation of NP occurrence(s).
Currently, no open, cross-kingdom, comprehensive and computer-interpretable electronic NP resource links NP and their containing organisms, along with referral to the underlying experimental work.
This shortcoming breaks the crucial evidentiary link required for tracing information back to the original data and assessing its quality.
Even valuable commercially available efforts for compiling NP data, such as the [Dictionary of Natural Products](https://dnp.chemnetbase.com) (DNP), can lack proper documentation of these critical links.

Pioneering efforts to address such challenges led to the establishment of [KNApSAck](http://www.knapsackfamily.com/KNApSAcK) [@doi:10.1007/3-540-29782-0_13], which is likely the first public, curated electronic NP resource of referenced structure-organism pairs.
KNApSAck currently contains 50,000+ structures and 100,000+ structure-organism pairs.
However, the organism field is not standardized and access to the data is not straightforward.
Another early-established electronic NP resources is the NAPRALERT dataset [@doi:10.1016/b978-008045382-8.00060-5], which was compiled over five decades from the NP literature, gathering and annotating data derived from over 200,000 primary literature sources.
This dataset contains 200,000+ distinct compound names and structural elements, along with 500,000+ records of distinct, fully-cited structure-organism pairs.
In total, NAPRALERT contains over 900,000 such records, due to equivalent structure-organism pairs reported in different citations.
However, NAPRALERT is not an open platform and employs an access model that provides only limited free searches of the dataset.
Finally, the [NPAtlas](https://www.npatlas.org) [@doi:10/gh5ptm; @doi:10.1093/nar/gkab941] is a more recent project complying with the FAIR (<u>F</u>indability, <u>A</u>ccessibility, <u>I</u>nteroperability and <u>R</u>euse) guidelines for digital assets [@doi:10/bdd4] and offering convenient web access.
While the NPAtlas allows retrieval and encourages submission of compounds with their biological source, it focuses on microbial NP and ignores a wide range of biosynthetically active organisms found in the Plantae kingdom.

The LOTUS initiative seeks to address the aforementioned shortcomings.
Building on the experience gained through the establishment of the recently published [COlleCtion of Open NatUral producTs](https://coconut.naturalproducts.net) (COCONUT) [@doi:10/ghssx5] regarding the aggregation and curation of NP structural databases, this _savoir-faire_ was expanded to accommodate biological organisms and scientific references in the equation.
After extensive data curation and harmonization of over 40 electronic ressources, pairs characterizing a NP occurrence were standardized at the chemical, biological and reference levels.
At its current stage of development, LOTUS disseminates 750,000+ referenced structure-organism pairs.
These efforts and experiences represent an intensive preliminary curatorial phase and the first major step towards providing a high quality, computer-interpretable knowledge base capable of transforming NP research data management from a classical (siloed) database approach to an optimally-shared resource.

### Accommodating Principles of FAIRness and TRUSTworthiness for Natural Products Knowledge Management

In awareness of the multi-faceted pitfalls associated with implementing, using and maintaining classical scientific DBs [@doi:10/gc4h7x], and to enhance current and future sharing options, the LOTUS initiative selected the [Wikidata](https://www.wikidata.org/wiki/Wikidata:Main_Page) platform for disseminating its resources.
The idea of using wikis to disseminate databases is not new, with multiple underlying advantages [@doi:10.1093/nar/gkr1195].
Since its creation, Wikidata has focused on cross-disciplinary and multilingual support.
Wikidata is curated and governed collaboratively by a global community of volunteers, about 20,000 of which are contributing monthly.
Wikidata currently contains more than 1 billion statements in the form of subject-predicate-object triples.
Triples are machine-interpretable and can be enriched with qualifiers and references.
Within Wikidata, data triples correspond to approximately 100 million entries, which can be grouped into classes as diverse as countries, songs, disasters, or chemical compounds.
The statements are closely integrated with [Wikipedia](https://en.wikipedia.org/wiki/Main_Page) and serve as the source for many of its infoboxes.
Various workflows have been established for reporting such classes, particularly those of interest to life sciences, such as genes, proteins, diseases, drugs, or biological taxa [@doi:10/ggqqc6].

Building on the principles and experiences described above, the present report introduces the development and implementation of the LOTUS workflow for NP occurrence curation and dissemination, which applies both FAIR and TRUST (<u>T</u>ransparency, <u>R</u>esponsibility, <u>U</u>ser focus, <u>S</u>ustainability and <u>T</u>echnology) principles [@doi:10/ggwrtj].
LOTUS data upload and retrieval procedures ensure optimal accessibility by the research community, allowing any researcher to contribute, edit and reuse the data with a clear and open CC0 license ([Creative Commons 0](https://creativecommons.org/publicdomain/zero/1.0)).

Despite many advantages, Wikidata hosting has some notable, yet manageable drawbacks.
While its SPARQL query language offers a powerful way to query available data, it can also appear intimidating to the less experienced user.
Furthermore, some typical queries of molecular electronic NP resources such as structural or spectral searches are not yet available in Wikidata.
To address these shortcomings, LOTUS is hosted in parallel at [https://lotus.naturalproducts.net](https://lotus.naturalproducts.net) (LNPN) within the naturalproducts.net ecosystem.
The [Natural Products Online](https://naturalproducts.net) website is a portal for open-source and open-data resources for NP research.
In addition to the generalistic COCONUT and LNPN databases, the portal will enable hosting of arbitrary and skinned collections, themed in particular by species or taxonomic clade, by geographic location or by institution, together with a range of cheminformatics tools for NP research.
LNPN is periodically updated with the latest LOTUS data.
This dual hosting provides an integrated, community-curated and vast knowledge base (*via* Wikidata), as well as a NP community-oriented product with tailored search modes (*via* LNPN).
The multiple data interaction options should establish the basis for the transparent and sustainable access, sharing and creation of knowledge on NP occurrence.


The LOTUS initiative was initially launched to answer our need to access the most comprehensive compilation of biological occurrences of NP.
Indeed, we recently highlighted the interest of considering the taxonomic dimension when annotating metabolites [@doi:10/fm73].
This being said, many other concrete applications can result from an access by the scientific community to the LOTUS initiative data.
For example, such a resource will facilitate the exploration of eco-evolutionary mechanisms at the molecular level [@doi:10/ghssx4].
In terms of drug discovery, this resource is extremely valuable to orient and guide researchers interested in a structure of interest.
On the same theme, LOTUS is expected to be the perfect place to encounter "molecular arguments" for biodiversity conservation [@doi:10.1046/j.1365-2664.2003.00803.x].
Researchers interested in the history of science will be able, through this kind of resource, to gain a preliminary view of the temporal evolution of disciplines such as pharmacognosy.
More generally, the objective of the LOTUS initiative is to prepare the ground for an electronic and globally accessible resource that would be the counterpart, at the metabolite level, of established databases linking proteins to biological organisms (eg. [Uniprot](https://www.uniprot.org/)) and genes to biological organisms ([Genbank](https://www.ncbi.nlm.nih.gov/genbank/)).
Once such an objective is reached, it will be possible to interconnect the three central objects of the living, i.e. metabolites, proteins and genes, through the common entity of these resources, the biological organism.
Such an interconnection, fostering cross-fertilization of the fields of chemistry, biology and associated disciplines is desirable and necessary to advance towards a better understanding of Life.


## Results & Discussion

This section is structured as follows: first, we present an [overview of the LOTUS initiative at its current stage of development](#blueprint-of-the-lotus-initiative).
The central [curation](#data-processing-&-validation) and [dissemination](#data-dissemination) elements of the LOTUS initiative are then explained in detail.
The [third section](#user-interaction-with-lotus-data) addresses the interaction modes between end-users and LOTUS, including _data retrieval_, _addition_ and _editing_.
Some examples on how LOTUS data can be used to answer research questions or develop hypothesis are given.
The [final section](#data-interpretation) is dedicated to the interpretation of LOTUS data and illustrates the dimensions and qualities of the current LOTUS dataset from chemical and biological perspectives.

### Blueprint of the LOTUS Initiative {.page_break_before}

Building on the standards established by the related WikiProjects on Wikidata ([Chemistry](https://www.wikidata.org/wiki/Wikidata:WikiProject_Chemistry), [Taxonomy](https://www.wikidata.org/wiki/Wikidata:WikiProject_Taxonomy) and [Source Metadata](https://www.wikidata.org/wiki/Wikidata:WikiProject_Source_MetaData)), a NP chemistry-oriented subproject was created ([Chemistry/Natural products](https://www.wikidata.org/wiki/Wikidata:WikiProject_Chemistry/Natural_products)).
Its central data consists of three minimal sufficient objects: 

- A *chemical structure object*, with associated Simplified Molecular Input Line Entry System (SMILES) [@doi:10/fg6q5f], International Chemical Identifier (InChI) [@doi:10/6bg] and InChIKey (a hashed version of the InChI).
- A *biological organism object*, with associated taxon name, the taxonomic DB where it was described and the taxon ID in the respective DB.
- A *reference object* describing the structure-organism pair, with the associated article title and a Digital Object Identifier (DOI), a PubMed (PMID), or PubMed Central (PMCID) ID.

As data formats are largely inhomogeneous among existing electronic NP resources, fields related to chemical structure, biological organism and references are variable and essentially not standardized.
Therefore, LOTUS implements multiple stages of harmonization, processing and validation (Figure @fig:workflow, stages 1 to 3).
LOTUS employs a Single Source of Truth (SSOT, [Single_source_of_truth](https://en.wikipedia.org/wiki/Single_source_of_truth)) to ensure data reliability and continuous availability of the latest curated version of LOTUS data in both Wikidata and LNPN (Figure @fig:workflow, stage 4).
The SSOT approach consists of a [PostgreSQL](https://www.postgresql.org/) DB that structures links and data schemes such that every data element has a single place.
The LOTUS processing pipeline is tailored to efficiently include and diffuse novel or curated data directly from new sources or at the Wikidata level.
This iterative workflow relies both on data addition and retrieval actions as described in the [Data Interaction](#user-interaction-with-lotus-data) section.
The overall process leading to referenced and curated structure-organisms pairs is illustrated in Figure @fig:workflow and detailed below.

\

![**Blueprint of the LOTUS initiative.** Data undergo a four-stage process: (1) Harmonization, (2) Processing, (3) Validation and (4) Dissemination. The process was designed to incorporate future contributions (5), either by the addition of new data from within Wikidata (a) or new sources (b) or *via* curation of existing data (c). The figure is available under the CC0 license at [https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_blueprint.svg](https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_blueprint.svg).](images/workflow.svg "Workflow"){#fig:workflow width="100%"}

By design, this iterative process fosters community participation, essential to efficiently document NP occurrences.
All stages of the workflow are described on the git sites of the LOTUS initiative at [https://github.com/lotusnprod](https://github.com/lotusnprod) and in the [methods](#Methods).
At the time of writing, 750,000+ LOTUS entries contained a curated chemical structure, biological organism and reference and were available on both Wikidata and LNPN.
As the LOTUS data volume is expected to increase over time, a frozen (as of 2021-12-20), tabular version of this dataset with its associated metadata is made available at [https://doi.org/10.5281/zenodo.5794106](https://doi.org/10.5281/zenodo.5794106) [@doi:10.5281/zenodo.5794106].

### Data Harmonization

Multiple data sources were processed as described hereafter.
All publicly accessible electronic NP resources included in [COCONUT](https://zenodo.org/record/3778405) that contain referenced structure-organism pairs were considered as initial input.
The data were complemented with COCONUT's own referenced structure-organism pairs [@doi:10/gh6dqw], as well as the following additional electronic NP resources: Dr. Duke [@DepartmentAgricultureAgricultural1992], Cyanometdb [@doi:10/gjgdzm], Datawarrior [@doi:10/f637rz], a subset of NAPRALERT, Wakankensaku [@MainPageWAKANKENSAKU] and DiaNat-DB [@doi:10.1039/D0RA10453A].

The contacts of the electronic NP resources not explicitly licensed as open were individually reached for permission to access and reuse data.
A detailed list of data sources and related information is available as [Appendix 1](#appendix-1-data-sources-list).
All necessary scripts for data gathering and harmonization can be found in the [lotus-processor](https://github.com/lotusnprod/lotus-processor) repository in the [src/1_gathering](https://github.com/lotusnprod/lotus-processor/tree/main/src/1_gathering) directory and processed is detailed in the corresponding methods section [gathering section](#data-gathering).
All subsequent iterations including new data sources, either updated information from the same data sources or new data, will involve a comparison with the previously gathered data at the SSOT level to ensure that the data is only curated once.

### Data Processing & Validation

As shown in Figure @fig:workflow, data curation consisted of three stages: harmonization, processing and validation.
Thereby, after the harmonization stage, each of the three central objects - chemical compounds, biological organisms and reference - were processed, as described in related [methods section](#data-processing).
Given the data size (2.5M+ initial entries), manual validation was unfeasible.
Curating the references was a particularly challenging part of the process.
Whereas organisms are typically reported by at least their vernacular or scientific denomination and chemical structures via their SMILES, InChI, InChIKey or image (not covered in this work), references suffer from largely insufficient reporting standards.
Despite poor standardization of the initial reference field, proper referencing remains an indispensable way to establish the validity of structure-organism pairs.
Better reporting practices, supported by tools such as [Scholia](https://scholia.toolforge.org) [@doi:10.5281/ZENODO.1036595; @doi:10.3897/rio.5.e35820] and relying on Wikidata, [Fatcat](https://fatcat.wiki), or [Semantic Scholar](https://www.semanticscholar.org) should improve reference-related information retrieval in the future.

In addition to curating the entries during data processing, 420 referenced structure-organism pairs were selected for manual validation.
An entry was considered as valid if: *i)* the structure (in the form of any structural descriptor that could be linked to the final sanitized InChIKey) was described in the reference *ii)* the containing organism (as any organism descriptor that could be linked to the accepted canonical name) was described in the reference and *iii)* the reference was describing the occurrence of the chemical structure in the biological organism.
More details are available in the related [methods section](#data-validation).
This process allowed us to establish rules for automatic filtering and validation of the entries.
The parameters of the automatic filtering are available as a function ([filter_dirty.R](https://github.com/lotusnprod/lotus-processor/blob/main/src/r/filter_dirty.R)) and are further described in the related [methods section](#data-validation).
The automatic filtering was then applied to all entries.
To confirm the efficacy of the filtering process, a new subset of 100 diverse, automatically curated and automatically validated entries was manually checked, yielding a rate of 97% of true positives.
The detailed results of the two manual validation steps are reported in Supporting Information [Appendix 2](#appendix-2-summary-of-the-validation-statistics).
The resulting data is also available in the dataset shared at [https://doi.org/10.5281/zenodo.5794106](https://doi.org/10.5281/zenodo.5794106) [@doi:10.5281/zenodo.5794106].
Table @tbl:examples shows an example of a referenced structure-organism pair before and after curation.
This process resolved the structure to an InChIKey, the organism to a valid taxonomic name and the reference to a DOI, thereby completing the establishment of the essential referenced structure-organism pair.

\

Table: Example of a referenced structure-organism pair before and after curation {#tbl:examples}

| | Structure | Organism | Reference | 
|:--------:|:-------------:|:-------------:|:-------------:|
| **Before curation** | Cyathocaline | Stem bark of Cyathocalyx zeylanica CHAMP. ex HOOK. f. & THOMS. (Annonaceae) | Wijeratne E. M. K., de Silva L. B., Kikuchi T., Tezuka Y., Gunatilaka A. A. L., Kingston D. G. I., J. Nat. Prod., 58, 459-462 (1995). | 
| **After curation** | VFIIVOHWCNHINZ-UHFFFAOYSA-N | Cyathocalyx zeylanicus | 10.1021/NP50117A020 | 

Challenging examples encountered during the development of the curation process were compiled in an edge cases table ([tests/tests.tsv](https://github.com/lotusnprod/lotus-processor/blob/main/tests/tests.tsv)) to allow for automated unit testing.
These tests allow a continuous revalidation of any change made to the code, ensuring that corrected errors will not reappear.
The alluvial plot in Figure @fig:alluvial illustrates the individual contribution of each *source* and *original* *subcategory* that led to the *processed* *categories*: structure, organism and reference.

\

![**Alluvial plot of the data transformation flow within LOTUS during the automated curation and validation processes.** The figure also reflects the relative proportions of the data stream in terms of the contributions from the various sources ("source" block, left), the composition of the harmonized subcategories ("original subcategory" block, middle) and the validated data after curation ("processed category" block, right). Automatically validated entries are represented in green, rejected entries in blue. The figure is available under the CC0 license at [https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_alluvial_plot.svg](https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_alluvial_plot.svg).](images/alluvial.svg "Alluvial"){#fig:alluvial width="100%"}

The figure highlights, for example, the essential contribution of the reference DOI category to the final validated entries.
A similar pattern can be seen concerning structures, where the validation rate of structural identifiers is higher than chemical names.
The combination of the results of the automated curation pipeline and the manually curated entries led to the establishment of four categories (manually validated, manually rejected, automatically validated and automatically rejected) of the referenced structure-organism pairs that formed the processed part of the SSOT.
Out of a total of 2.5M+ initial pairs, the manual and automatic validation retained 750,000+ pairs (approximately 30%), which were then selected for dissemination on Wikidata.
Among validated entries, multiple ones were redundant among the source databases, thus also explaining the decrease of entries between the initial pairs and validated ones.
Moreover, because data quality was favored over quantity, the number of rejected entries is high.
Among them, multiple correct entries were certainly falsely rejected, but still not disseminated.
All rejected entries were kept aside for later manual inspection and validation.
These are publicly available at [https://doi.org/10.5281/zenodo.5794597](https://doi.org/10.5281/zenodo.5794597) [@doi:10.5281/zenodo.5794597].
In the end, the disseminated data contained 290,000+ unique chemical structures, 40,000+ distinct organisms and 75,000+ references.

### Data Dissemination

Research worldwide can benefit the most when all results of published scientific studies are fully accessible immediately upon publication [@doi:10.1038/417222b].
This concept is considered the foundation of scientific investigation and a prerequisite for the effective direction of new research efforts based on prior information.
To achieve this, research results have to be made publicly available and reusable.
As computers are now the main investigation tool for a growing number of scientists, all research data including those in publications should be disseminated in computer-readable format, following the FAIR principles.
LOTUS uses Wikidata as a repository for referenced structure-organism pairs, as this allows documented research data to be integrated with a large, pre-existing and extensible body of chemical and biological knowledge.
The dynamic nature of Wikidata encourages the continuous curation of deposited data through the user community.
Independence from individual and institutional funding represents another major advantage of Wikidata.
The Wikidata knowledge base and the option to use elaborate SPARQL queries allow the exploration of the dataset from a sheer unlimited number of angles.
The openness of Wikidata also offers unprecedented opportunities for community curation, which will support, if not guarantee, a dynamic and evolving data repository.
At the same time, certain limitations of this approach can be anticipated.
Despite (or possibly due to) their power, SPARQL queries can be complex and potentially require an in-depth understanding of the models and data structure.
This involves a steep learning curve which can discourage some end-users.
Furthermore, traditional ways to query electronic NP resources such as structural or spectral searches are currently not within the scope of Wikidata and, are thus addressed in LNPN.
Using the pre-existing COCONUT template, LNPN hosting allows the user to perform structural searches by directly drawing a molecule, thereby addressing the current lack of such structural search possibilities in Wikidata.
Since metabolite profiling by Liquid Chromatography (LC) - Mass Spectrometry (MS) is now routinely used for the chemical composition assessment of natural extracts, future versions of LOTUS and COCONUT are envisioned to be augmented by predicted MS spectra and hosted at [https://naturalproducts.net](https://naturalproducts.net) to allow mass and spectral-based queries.
Note that such spectral database is already available at [https://doi.org/10.5281/zenodo.5607264](https://doi.org/10.5281/zenodo.5607264) [@doi:10.5281/zenodo.5607264].
To facilitate queries focused on specific taxa (e.g., "return all molecules found in the Asteraceae family"), a unified taxonomy is paramount.
As the taxonomy of living organisms is a complex and constantly evolving field, all the taxon identifiers from all accepted taxonomic DB for a given taxon name were kept.
Initiatives such as the [Open Tree of Life](https://tree.opentreeoflife.org) (OTL) [@doi:10/ggkjbt] will help to gradually reduce these discrepancies, and the Wikidata platform can and does support such developments.
OTL also benefits from regular expert curation and new data.
As the taxonomic identifier property for this resource did not exist in Wikidata, its creation was requested and obtained.
The property is now available as "Open Tree of Life ID" ([P9157](https://www.wikidata.org/wiki/Property:P9157)).

Following the previously described curation process, all validated entries have been made available through Wikidata and LNPN.
LNPN will be regularly mirroring Wikidata LOTUS through the SSOT as described in Figure @fig:workflow.

\

### User Interaction with LOTUS Data

The possibilities to interact with the LOTUS data are numerous.
The following gives examples of how to retrieve, add and edit LOTUS data.

#### Data Retrieval

LOTUS data can be queried and retrieved either directly in Wikidata or on LNPN, both of which have distinct advantages.
While Wikidata offers flexible and powerful queries capacities at the cost of potential complexity, LNPN has a graphical user interface with capabilities of drawing chemical structures, simplified structural or biological filtering and advanced chemical descriptors, albeit with a more rigid structure.
For bulk download, a frozen version of LOTUS data (timestamp of 2021-12-20) is also available at [https://doi.org/10.5281/zenodo.5794106](https://doi.org/10.5281/zenodo.5794106) [@doi:10.5281/zenodo.5794106].
More refined approaches to the direct interrogation of the up-to-date LOTUS data both in Wikidata and LNPN are detailed hereafter.

##### Wikidata

The easiest way to search for NP occurrence information in Wikidata is by typing the name of a chemical structure directly into the "Search Wikidata" field, which (for left-to-right languages) can be found in the upper right corner of the [Wikidata homepage](https://www.wikidata.org/wiki/Wikidata:Main_Page) or any other Wikidata page.
For example, by typing "erysodine", the user will land on the page of this compound ([Q27265641](https://www.wikidata.org/wiki/Q27265641)).
Scrolling down to the "found in taxon" statement will allow the user to view the biological organisms reported to contain this NP (Figure @fig:wdpage).
Clicking the reference link under each taxon name links to the publication(s) documenting the occurrence.

![**Illustration of the "found in taxon" statement section on the Wikidata page** of erysodine [Q27265641](https://www.wikidata.org/wiki/Q27265641) showing a selection of erysodine-containing taxa and the references documenting these occurrences.](images/erysodine.svg "wdpage"){#fig:wdpage width="100%"}

The typical approach to more elaborated querying involves writing SPARQL queries using the [Wikidata Query Service](https://query.wikidata.org) or another direct connection to a SPARQL endpoint.
Below are some examples from simple to more elaborated queries, demonstrating what can be done using this approach.
The full-text queries with explanations are included in [Appendix 3](#appendix-3-wikidata-queries).

Table: Potential questions about structure-organism relationships and corresponding Wikidata queries. {#tbl:queries}

| Question | Wikidata SPARQL query | 
|:------------------------:|:-------------:|
| What are the compounds present in Mouse-ear cress (*Arabidopsis thaliana*) or its child taxa? | [https://w.wiki/4Vcv](https://w.wiki/4Vcv) | 
| Which organisms are known to contain *β*-sitosterol? | [https://w.wiki/4VFn](https://w.wiki/4VFn) | 
| Which organisms are known to contain stereoisomers of *β*-sitosterol?| [https://w.wiki/4VFq](https://w.wiki/4VFq) | 
| Which pigments are found in which taxa, according to which reference? | [https://w.wiki/4VFx](https://w.wiki/4VFx) |
| What are examples of organisms where compounds were found in an organism sharing the same parent taxon, but not in the organism itself? | [https://w.wiki/4Wt3](https://w.wiki/4Wt3) |
| Which *Zephyranthes* species lack compounds known from at least two species in the genus? | [https://w.wiki/4VG3](https://w.wiki/4VG3) |
| How many compounds are structurally similar to compounds labeled as antibiotics? (grouped by the parent taxon of the containing organism) | [https://w.wiki/4VG4](https://w.wiki/4VG4) |
| Which organisms contain indolic scaffolds? Count occurrences, group and order the results by the parent taxon. | [https://w.wiki/4VG9](https://w.wiki/4VG9) |
| Which compounds with known bioactivities were isolated from Actinobacteria, between 2014 and 2019, with related organisms and references? | [https://w.wiki/4VGC](https://w.wiki/4VGC) |
| Which compounds labeled as terpenoids were found in *Aspergillus* species, between 2010 and 2020, with related references? | [https://w.wiki/4VGD](https://w.wiki/4VGD) |
| Which are the available referenced structure-organism pairs on wikidata? (example limited to 1000 results) | [https://w.wiki/4VFh](https://w.wiki/4VFh) |

The queries presented in Table @tbl:queries are only selected examples, and many other ways of interrogating LOTUS can be formulated.
Generic queries can be used, for example, for hypothesis generation when starting a research project.
For instance, a generic SPARQL query - listed in Table @tbl:queries as "Which are the available referenced structure-organism pairs on wikidata?" - retrieves all structures, identified by their InChIKey ([P235](https://www.wikidata.org/wiki/Property:P235)), which contain "found in taxon" ([P703](https://www.wikidata.org/wiki/Property:P703)) statements that are stated in ([P248](https://www.wikidata.org/wiki/Property:P248)) a bibliographic reference: [https://w.wiki/4VFh](https://w.wiki/4VFh).
Data can then be exported in various formats, such as classical tabular formats, json, or html tables (see Download tab on the lower right of the query frame).
At the time of writing (2021-12-20), this query (without the LIMIT 1000) returned 951,800 entries; a frozen query result is available at [https://doi.org/10.5281/zenodo.5668854](https://doi.org/10.5281/zenodo.5668854) [@doi:10.5281/zenodo.5668854].

Targeted queries allowing to interrogate LOTUS data from the perspective of one of the three objects forming the referenced structure-organism pairs can also be built.
Users can, for example, retrieve a list of all structures reported from a given organism, such as all structures reported from *Arabidopsis thaliana* ([Q158695](https://www.wikidata.org/wiki/Q158695)) or its child taxa ([https://w.wiki/4Vcv](https://w.wiki/4Vcv)).
Alternatively, all organisms containing a given chemical can be queried via its structure, such as in the search for all organisms where *β*-sitosterol ([Q121802](https://www.wikidata.org/wiki/Q121802)) was found in ([https://w.wiki/4VFn](https://w.wiki/4VFn)).
For programmatic access, the [lotus-wikidata-exporter](https://github.com/lotusnprod/lotus-wikidata-exporter) repository also allows data retrieval in RDF format and as TSV tables.

To further showcase the possibilities, two additional queries were established ([https://w.wiki/4VGC](https://w.wiki/4VGC) and [https://w.wiki/4VGD](https://w.wiki/4VGD)).
Both queries were inspired by recent literature review works [@doi:10.1016/j.micres.2021.126708; @doi:10.1016/j.phytochem.2021.113011].
The first work describes compounds found in Actinobacteria, with a biological focus on compounds with reported bioactivity.
The second one describes compounds found in *Aspergillus* spp., with a chemical focus on terpenoids.
In both cases, in seconds, the queries allow retrieving a table similar to the ones in the mentioned literature reviews.
While these queries are not a direct substitute for manual literature review, they do allow researchers to quickly begin such a review process with a very strong body of relevant references.

For a convenient expansion or limitation of the results, certain types of queries such as structure or similarity searches usually exist in molecular electronic resources.
As these queries are not natively integrated by SPARQL, they are not readily available for Wikidata exploration.
To address such limitation, Galgonek et al. developed an in-house SPARQL engine that allows utilization of Sachem, a high-performance chemical DB cartridge for PostgreSQL for fingerprint-guided substructure and similarity search [@doi:10/gdn52j].
The engine is used by the Integrated Database of Small Molecules (IDSM) that operates, among other things, several dedicated endpoints allowing structural search in selected small-molecule datasets via SPARQL [@doi:10/gf5bv5].
To allow substructure and similarity searches via SPARQL also on compounds from Wikidata, a [dedicated IDSM/Sachem endpoint](https://idsm.elixir-czech.cz/sparql/endpoint/wikidata) was created for the LOTUS project.
The endpoint indexes isomeric ([P2017](https://www.wikidata.org/wiki/Property:P2017)) and canonical ([P233](https://www.wikidata.org/wiki/Property:P233)) SMILES code available in Wikidata.
To ensure that data is kept up-to-date, SMILES codes are automatically downloaded from Wikidata daily.
The endpoint allows users to run [federated queries](https://www.w3.org/TR/sparql11-federated-query) and, thereby, proceed to structure-oriented searches on the LOTUS data hosted at Wikidata.
For example, the SPARQL query [https://w.wiki/4VG9](https://w.wiki/4VG9) returns a list of all organisms that produce NP with an indolic scaffold.
The output is aggregated at the parent taxa level of the containing organisms and ranked by the number of scaffold occurrences.

Regarding the versioning aspects, some challenges are implied by the dynamic nature of the Wikidata environment.
However, tracking of the data evolution can be achieved in multiple ways and at different levels: at the full Wikidata level, dumps are regularly created ([https://dumps.wikimedia.org/wikidatawiki/entities](https://dumps.wikimedia.org/wikidatawiki/entities)) while at the individual entry level the full history of modification can be consulted (see following link for the full edit history of erythromycin for example ([https://www.wikidata.org/w/index.php?title=Q213511&action=history](https://www.wikidata.org/w/index.php?title=Q213511&action=history))).

We propose to the users a simple approach to document, version and share the output of queries on the LOTUS data at a defined time point.
For this, in addition to sharing the short url of a the SPARQL query (which will return results evolving over time), a simple archiving of the returned table to Zenodo or similar platform can be done.
In order to gather results of SPARQL queries we established [the LOTUS Initiative Community repository](https://zenodo.org/communities/the-lotus-initiative/).
The following link allows to directly contribute to the community repository [https://zenodo.org/deposit/new?c=the-lotus-initiative](https://zenodo.org/deposit/new?c=the-lotus-initiative).
For example the output of this Wikidata SPARQL query [https://w.wiki/4N8G](https://w.wiki/4N8G) realized on the 2021-11-10T16:56 can be easily archived and shared in a publication via its DOI [10.5281/zenodo.5668380](https://doi.org/10.5281/zenodo.5668380).

##### Lotus.NaturalProducts.Net (LNPN)

In the search field of the LNPN interface ([https://lotus.naturalproducts.net](https://lotus.naturalproducts.net)), simple queries can be achieved by typing the molecule name (e.g., [ibogaine](https://lotus.naturalproducts.net/search/simple/ibogaine)) or pasting a SMILES, InChI, InChIKey string, or a Wikidata identifier.
All compounds reported from a given organism can be found by entering the organism name at the species or any higher taxa level (e.g. [*Tabernanthe iboga*](https://lotus.naturalproducts.net/search/simple/Tabernanthe%20iboga)).
Compound search by chemical class is also possible.

Alternatively, a structure can be directly drawn in the structure search interface ([https://lotus.naturalproducts.net/search/structure](https://lotus.naturalproducts.net/search/structure)), where the user can also decide on the nature of the structure search (exact, similarity, substructure search).
Refined search mode combining multiple search criteria, in particular physicochemical properties, is available in the advanced search interface ([https://lotus.naturalproducts.net/search/advanced](https://lotus.naturalproducts.net/search/advanced)).

Within LNPN, LOTUS bulk data can be retrieved as SDF or SMILES files, or as a complete MongoDB dump via [https://lotus.naturalproducts.net/download](https://lotus.naturalproducts.net/download).
Extensive documentation describing the search possibilities and data entries is available at [https://lotus.naturalproducts.net/documentation](https://lotus.naturalproducts.net/documentation).
LNPN can also be queried via the application programming interface (API) as described in the documentation.

#### Data Addition and Evolution {.page_break_before}

One major advantage of the LOTUS architecture is that every user has the option to contribute to the NP occurrences documentation effort by adding new or editing existing data.
As all LOTUS data applies the SSOT mechanism, reprocessing of previously treated elements is avoided.
However, at the moment, the SSOT channels are not open to the public for direct write access to maintain data coherence and evolution of the SSOT scheme.
For now, the users can employ the following approaches to add or modify data in LOTUS.

##### Sources

LOTUS data management involves regular re-importing of both current and new data sources.
New and edited information from these electronic NP resources will be checked against the SSOT.
If absent or different, data will be passed through the curation pipeline and subsequently stored in the SSOT.
Accordingly, by contributing to external electronic NP resources, any researcher has a means of providing new data for LOTUS, keeping in mind the inevitable delay between data addition and subsequent inclusion into LOTUS.

##### Wikidata

The currently favored approach to add new data to LOTUS is to create or edit Wikidata entries directly.
Newly created or edited data will then be imported into the SSOT.
There are several ways to interact with Wikidata which depend on the technical skills of the user and the volume of data to be uploaded/modified.

###### Pre-requisites

While direct Wikidata upload is possible, contributors are encouraged to use the LOTUS curation pipeline as a preliminary step to strengthen the initial data quality.
For this, a specific mode of the LOTUS processor can be called (see [Custom mode](https://github.com/lotusnprod/lotus-processor/wiki/Make-commands#custom)).
The added data will therefore benefit from the curation and validation stages implemented in the LOTUS processing pipeline.

###### Manual Upload

Any researcher interested in reporting NP occurrences can manually add the data directly in Wikidata, without any particular technical knowledge requirement.
For this the creation of a Wikidata account and following the [general object editing guidelines](https://www.wikidata.org/wiki/Wikidata:Tours) is advised.
Regarding the addition of NP-centered objects (i.e., referenced structure-organisms pairs), users shall refer to the [WikiProject Chemistry/Natural products](https://www.wikidata.org/wiki/Wikidata:WikiProject_Chemistry/Natural_products) group page.

A tutorial for the manual creation and upload of a referenced structure-organism pair to Wikidata is available in [Appendix 4](#appendix-4-wikidata-entry-creation-tutorial).


###### Batch and Automated Upload

Through the initial curation process described previously, 750,000+ referenced structure-organism pairs were validated for Wikidata upload.
To automate this process, a set of programs were written to automatically process the curated outputs, group references, organisms and compounds, check if they are already present in Wikidata (using SPARQL and direct Wikidata querying) and insert or update the entities as needed (i.e., upserting).
These scripts can be used for future batch upload of properly curated and referenced structure-organism pairs to Wikidata.
Programs for data addition to Wikidata can be found in the repository [lotus-wikidata-interact](https://github.com/lotusnprod/lotus-wikidata-interact).
The following [Xtools page](https://xtools.wmflabs.org/ec/www.wikidata.org/NPImporterBot) offers an overview of the latest activity performed by our [NPimporterBot](https://www.wikidata.org/wiki/User:NPImporterBothttps://www.wikidata.org/wiki/User:NPImporterBot), using those programs.

#### Data Editing {.page_break_before}

Even if correct at a given time point, scientific advances can invalidate or update previously uploaded data.
Thus, the possibility to continuously edit the data is desirable and guarantees data quality and sustainability.
Community-maintained knowledge bases such as Wikidata encourage such a process.
Wikidata presents the advantage of allowing both manual and automated correction.
Field-specific robots such as [SuccuBot](https://www.wikidata.org/wiki/User:SuccuBot), [KrBot](https://www.wikidata.org/wiki/User:KrBot), [Pi_bot](https://www.wikidata.org/wiki/User:Pi_bot) and [ProteinBoxBot](https://www.wikidata.org/wiki/User:ProteinBoxBot) or our [NPimporterBot](https://www.wikidata.org/wiki/User:NPImporterBot) went through an approval process.
The robots are capable of performing thousands of edits without the need for human input.
This automation helps reduce the amount of incorrect data that would otherwise require manual editing.
However, manual curation by human experts remains irreplaceable as a standard.
Users who value this approach and are interested in contributing are invited to follow the manual curation tutorial in [Appendix 4](#appendix-4-wikidata-entry-creation-tutorial).

The [Scholia platform](https://scholia.toolforge.org) provides a visual interface to display the links among Wikidata objects such as researchers, topics, species or chemicals.
It now provides an interesting way to view the chemical compounds found in a given biological organism (see here for the metabolome view of [*Eurycoma longifolia*](https://scholia.toolforge.org/taxon/Q311710)).
If Scholia currently does not offer a direct editing interface for scientific references, it still allows users to proceed to convenient batch editing via [Quick Statements](https://www.wikidata.org/wiki/Help:QuickStatements).
The adaptation of such a framework to edit the referenced structure-pairs in the LOTUS initiative could thus facilitate the capture of future expert curation, especially manual efforts that cannot be replaced by automated scripts.

### Data Interpretation

To illustrate the nature and dimensions of the LOTUS dataset, some selected examples of data interpretation are shown.
First, the distribution of chemical structures among four important NP reservoirs: plants, fungi, animals and bacteria (Table @tbl:distribution).
Then, the distribution of biological organisms according to the number of related chemical structures and likewise the distribution of chemical structures across biological organisms are illustrated (Figure @fig:distribution).
Furthermore, the individual electronic NP resources participation in LOTUS data is resumed using the UpSet plot depiction, which allows the visualization of intersections in data sets (Figure @fig:upset).
Across these figures we take again the two previous examples, i.e, *β*-sitosterol as chemical structure and *Arabidopsis thaliana* as biological organism because of their well-documented statuses.
Finally, a biologically-interpreted chemical tree and a chemically-interpreted biological tree are presented (Figure @fig:tmap and Figure @fig:magicTree).
The examples illustrate the overall chemical and biological coverage of LOTUS by linking family-specific classes of chemical structures to their taxonomic position.
Table @tbl:distribution, Figures @fig:distribution, @fig:tmap and @fig:magicTree were generated using the frozen data (2021-12-20 timestamp), which is available for download at [https://doi.org/10.5281/zenodo.5794106](https://doi.org/10.5281/zenodo.5794106) [@doi:10.5281/zenodo.5794106].
Figure @fig:upset required a dataset containing information from closed resources and the complete data used for its generation is therefore not available for public distribution.
All scripts used for the generation of the figures are available in the [lotus-processor](https://github.com/lotusnprod/lotus-processor) repository in the [src/4_visualizing](https://github.com/lotusnprod/lotus-processor/tree/main/src/4_visualizing) directory for reproducibility.

#### Distribution of Chemical Structures across reported Biological Organisms in LOTUS

Table @tbl:distribution summarizes the distribution of chemical structures and their chemical classes (according to NPClassifier [@doi:10.1021/acs.jnatprod.1c00399]) across the biological organisms reported in LOTUS.
For this, biological organisms were grouped into four artificial taxonomic levels (plants, fungi, animals and bacteria).
These were built by combining the two highest taxonomic levels in the OTL taxonomy, namely Domain and Kingdom levels.
"Plants" corresponded to "Eukaryota_Archaeplastida", "Fungi" to "Eukaryota_Fungi", "Animals" to "Eukaryota_Metazoa" and "Bacteria" to "Bacteria_NA".
The category corresponding to "Eukaryota_NA" mainly contained Algae, but also other organisms such as Amoebozoa and was therefore excluded.
This represented less than 1% of all entries.
The details of this process are available under [src/3_analyzing/structure_taxon_distribution.R](https://github.com/lotusnprod/lotus-processor/blob/main/src/3_analyzing/structure_taxon_distribution.R).
When the chemical structure/class was reported only in one taxonomic grouping, it was counted as "specific".

Table: Distribution and specificity of chemical structures across four important NP reservoirs: plants, fungi, animals and bacteria. When the chemical structure/class appeared only in one group and not the three others, they were counted as "specific". Chemical classes were attributed with NPClassifier. {#tbl:distribution}

| **Group** | **Organisms** | **2D Structure-Organism Pairs** | **2D Chemical Structures** | **Specific Chemical Structures** | **Chemical Classes** | **Specific Chemical Classes** | 
|:-------------:|-----------:|--------------:|-------------:|--------------:|-------------:|-------------:|
| **Plantae** | 28,439 | 342,891 | 95,191 | 90,672 (95%) | 545 | 59 (11%) |
| **Fungi** | 4,003 | 36,950 | 22,594 | 20,194 (89%) | 417 | 19 (5%) |
| **Animalia** | 2,716 | 24,114 | 15,242 | 11,822 (78%) | 455 | 14 (3%) |
| **Bacteria** | 1,555 | 23,198 | 15,895 | 14,130 (89%) | 385 | 43 (11%) |

#### Distributions of Organisms per Structure and Structures per Organism 

Readily achievable outcomes from LOTUS show that the depth of exploration of the world of NP is rather limited: as depicted in Figure @fig:distribution, on average, three organisms are reported per chemical structure and eleven structures per organism.
Notably, half of all structures have been reported from a single organism and half of all studied organisms are reported to contain five or fewer structures.
Metabolomics studies suggest that these numbers are heavily underrated [@doi:10/b3z2g7; @doi:10/ggwr93] and indicate that a better reporting of the metabolites detected in the course of NP chemistry investigations should greatly improve coverage. 

This incomplete coverage may be partially explained by the habit in classical NP journals to accept only new and/or bioactive chemical structures for publication.
Another possible explanation is the fact that specific chemical classes have been under heavier scrutiny by the natural products community than others.
For example, alkaloids have three specific characteristics which favor their reporting in the literature.
First, they are often endowed with potent biological activities making them a target in the frame of pharmacognosy research.
Second, their chemical nature makes them readily accessible from complex biological matrices through acido-basic extraction.
Third, they ionize greatly in positive MS mode, which makes their detection even at a very low concentration possible, where other compounds present in much higher concentrations are not detected.
It is thus a complex task to answer the following question: "Is the currently observed repartition of alkaloids across the tree of life a reflection of their true biological occurrence or is this repartition biased by the aforementioned characteristics of this chemical class ?" 
While the LOTUS initiative does not allow yet disentangling the bias from the true occurrence, it should offer sound and strong foundations for such challenging research problematic.

Another obvious explanation for the limited coverage is the fact that most of the chemical structures in LOTUS have been physically isolated and described.
This is an extremely time-consuming effort that can obviously not be carried on all metabolites of all biological organisms.
Here, the sensitivity of mass spectrometry and the ever-increasing efficiency of computational metabolite annotation solutions could offer a strong take.
The documentation of metabolite annotation results obtained on large collections of biological matrices and the associated metadata within knowledge graphs offers exciting perspectives regarding the possibilities to expand both the chemical and biological coverage of the LOTUS data in a feasible manner.

![**Distribution of "structures per organism" and "organisms per structure"**. The number of organisms linked to the planar structure of *β*-sitosterol (KZJWDPNRJALLNS) and the number of chemical structures in *Arabidopsis thaliana are* two exemplary highlights. *A. thaliana* contains 687 different short InChIKeys (i.e 2D structures) and KZJWDPNRJALLNS is reported in 3979 distinct organisms. Less than 10 percent of the species contain more than 80 percent of the structural diversity present within LOTUS. In parallel, 80 percent of the species present in LOTUS are linked to less than 10 percent of the structures. The figure is available under the CC0 license at [https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_structure_organism_distribution.svg](https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_structure_organism_distribution.svg).](images/distribution.svg "distribution"){#fig:distribution width="100%"}

#### Contribution of Individual Electronic NP Resources to LOTUS {.page_break_before}

The added value of the LOTUS initiative to assemble multiple electronic NP resources is illustrated in Figure @fig:upset : Panel A shows the contributions of the individual electronic NP resources to the ensemble of chemical structures found in one of the most studied vascular plants, *Arabidopsis thaliana* ("Mouse-ear cress"; [Q147096](https://www.wikidata.org/wiki/Q147096)).
Panel B shows the ensemble of taxa reported to contain the planar structure of the widely occurring triterpenoid *β*-sitosterol ([Q121802](https://www.wikidata.org/wiki/Q121802)).

![**UpSet plots of the individual contribution of electronic NP resources** to the planar structures found in *Arabidopsis thaliana* (A) and to organisms reported to contain the planar structure of *β*-sitosterol (KZJWDPNRJALLNS) (B). UpSet plots are evolved Venn diagrams, allowing to represent intersections between multiple sets. The horizontal bars on the lower left represent the number of corresponding entries per electronic NP resource. The dots and their connecting line represent the intersection between source and consolidate sets. The vertical bars indicate the number of entries at the intersection. For example, 479 organisms containing the planar structure of *β*-sitosterol are present in both UNPD and NAPRALERT, whereas each of them respectively reports 1,349 and 2,330 organisms containing the planar structure of *β*-sitosterol. The figure is available under the CC0 license at [https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_upset_plot.svg](https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_upset_plot.svg).](images/upset_combined.svg "upset"){#fig:upset width="100%"}

Figure @fig:upset A. also shows that according to NPClassifier, the "chemical pathway" category distribution across electronic NP resources is not conserved.
Note that NPClassifier and ClassyFire [@doi:10/gc5tqv] chemical classification results are both available as metadata in the frozen LOTUS export and LNPN.
Both classification tools return a chemical taxonomy for individual structures, thus allowing their grouping at higher hierarchical levels, in the same way as it is done for biological taxonomies.
The UpSet plot in Figure @fig:upset indicates the poor overlap of preexisting electronic NP resources and the added value of an aggregated dataset.
This is particularly well illustrated in Figure @fig:upset B., where the number of organisms for which the planar structure of *β*-sitosterol (KZJWDPNRJALLNS) has been reported is shown for each intersection.
NAPRALERT has by far the highest number of entries (2,330 in total), while other electronic NP resources complement this well: e.g., UNPD has 573 reported organisms with *β*-sitosterol that do not overlap with any other resource.
Of note, *β*-sitosterol is documented in only 13 organisms in the DNP, highlighting the importance of a better systematic reporting of ubiquitous metabolites and the interest of multiple data sources agglomeration.

#### A Biologically-interpreted Chemical Tree

The chemical diversity captured in LOTUS is here displayed using TMAP (Figure @fig:tmap), a visualization library allowing the structural organization of large chemical datasets as a minimum spanning tree [@doi:10/gjjz4f].
Using Faerun, an interactive HTML file is generated to display metadata and molecule structures by embedding the SmilesDrawer library [@doi:10/gcmhc5; @doi:10/gjjz4d].
Planar structures were used for all compounds to generate the TMAP (chemical space tree-map) using MAP4 encoding [@doi:10/gjjz4c].
As the tree organizes structures according to their molecular fingerprint, an anticipated coherence between the clustering of compounds and the mapped NPClassifier chemical class is observed (Figure @fig:tmap A.).
For clarity, some of the most represented chemical classes of LOTUS plus quassinoids and stigmastane steroids are mapped, with examples of a quassinoid (NXZXPYYKGQCDRO) (light green star) and a stigmastane steroid (KZJWDPNRJALLNS) (dark green diamond) and their corresponding location in the TMAP.

![**TMAP visualizations of the chemical diversity present in LOTUS.** Each dot corresponds to a chemical structure. A highly specific quassinoid (NXZXPYYKGQCDRO) (light green star) and an ubiquitous stigmastane steroid (KZJWDPNRJALLNS) (dark green diamond) are mapped as examples in all visualizations. In panel A., compounds (dots) are colored according to the NPClassifier chemical class they belong to. In panel B., compounds that are mostly reported in the Simaroubaceae family are highlighted in blue. Finally, in panel C., the compounds are colored according to the specificity score of chemical classes found in biological organisms. This biological specificity score at a given taxonomic level for a given chemical class is calculated as a [Jensen-Shannon divergence](https://en.wikipedia.org/wiki/Jensen%E2%80%93Shannon_divergence). A score of 1 suggests that compounds are highly specific, 0 that they are ubiquitous. Zooms on a group of compounds of high biological specificity score (in pink) and on compounds of low specificity (blue) are depicted. An interactive HTML visualization of the LOTUS TMAP is available at [https://lotus.nprod.net/post/lotus-tmap/](https://lotus.nprod.net/post/lotus-tmap/) and archived at [https://doi.org/10.5281/zenodo.5801807](https://doi.org/10.5281/zenodo.5801807) [@doi:10.5281/zenodo.5801807]. The figure is available under the CC0 license at [https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_biologically_interpreted_chemical_tree.svg](https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_biologically_interpreted_chemical_tree.svg).](images/tmap.svg "tmap"){#fig:tmap width="100%"}

<br/><br/>
<br/><br/>

To explore relationships between chemistry and biology, it is possible to map taxonomical information such as the most reported biological family per chemical compound (Figure @fig:tmap B.) or the biological specificity of chemical classes (Figure @fig:tmap C.) on the TMAP.
The biological specificity score at a given taxonomic level for a given chemical class is calculated as a [Jensen-Shannon divergence](https://en.wikipedia.org/wiki/Jensen%E2%80%93Shannon_divergence).
A score of 1 suggests that compounds are highly specific, 0 that they are ubiquitous.
For more details, see [3_analyzing/jensen_shannon_divergence.R](https://github.com/lotusnprod/lotus-processor/blob/main/src/3_analyzing/jensen_shannon_divergence.R).
This visualization allows to highlight chemical classes specific to a given taxon, such as the quassinoids in the Simaroubaceae family.
In this case it is striking to see how well the compounds of a given chemical class (quassinoids) (Figure @fig:tmap A.) and the most reported plant family per compound (Simaroubaceae) (Figure @fig:tmap B.) overlap.
This is also evidenced on Figure @fig:tmap C. with a Jensen-Shannon divergence of 0.99 at the biological family level for quassinoids.
In this plot, it is also possible to identify chemical classes that are widely spread among living organisms, such as the stigmastane steroids, which exhibit a Jensen-Shannon divergence of 0.73 at the biological family level.
This means that repartition of stigmastane steroids among families is not specific.
Figure @fig:magicTree-additional further supports this statement.

#### A Chemically-interpreted Biological Tree

An alternative view of the biological and chemical diversity covered by LOTUS is illustrated in Figure @fig:magicTree.
Here, chemical compounds are not organized but biological organisms are placed in their taxonomy.
To limit bias due to under-reporting in the literature and keep a reasonable display size, only families with at least 50 reported compounds were included.
Organisms were classified according to the OTL taxonomy and structures according to NPClassifier.
The tips were labeled according to the biological family and colored according to their biological kingdom.
The bars represent structure specificity of the most characteristic chemical class of the given biological family (the higher the more specific).
This specificity score was a [Jaccard index](https://en.wikipedia.org/wiki/Jaccard_index) between the chemical class and the biological family.
For more details, see [4_visualizing/plot_magicTree.R](https://github.com/lotusnprod/lotus-processor/blob/main/src/4_visualizing/plot_magicTree.R).

![**LOTUS provides new means of exploring and representing chemical and biological diversity.** The tree generated from current LOTUS data builds on biological taxonomy and employs the kingdom as tips label color (only families containing 50+ chemical structures were considered). The outer bars correspond to the most specific chemical class found in the biological family. The height of the bar is proportional to a specificity score corresponding to a [Jaccard index](https://en.wikipedia.org/wiki/Jaccard_index) between the chemical class and the biological family. The bar color corresponds to the chemical pathway of the most specific chemical class in the NPClassifier classification system. The size of the leaf nodes corresponds to the number of genera reported in the family. The figure is vectorized and zoomable for detailed inspection and is available under the CC0 license at [https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_chemically_interpreted_biological_tree.svg](https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_chemically_interpreted_biological_tree.svg).](images/magicTree.svg "magicTree"){#fig:magicTree width="100%"}

![Distribution of *β*-sitosterol and related chemical parents among families with at least 50 reported compounds present in LOTUS. Script used for the generation of each tree in the figure is the same ([src/4_visualizing/plot_magicTree.R](https://github.com/lotusnprod/lotus-processor/blob/main/src/4_visualizing/plot_magicTree.R)) as for Figure @fig:magicTree as both figures are related. The figure is available under CC0 license at [https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_chemically_interpreted_biological_tree_supplement.svg](https://commons.wikimedia.org/wiki/File:Lotus_initiative_1_chemically_interpreted_biological_tree_supplement.svg).](images/magicTree_additional.svg "magicTree_additional"){#fig:magicTree-additional tag="7-figure supplement 1"}

Figure @fig:magicTree makes it possible to spot highly specific compound classes such as trinervitane terpenoids in the Termitidae, the rhizoxin macrolides in the Rhizopodaceae, or the quassinoids and limonoids typical, respectively, of Simaroubaceae and Meliaceae.
Similarly, tendencies of more generic occurrence of NP can be observed.
For example, within the fungal kingdom, Basidiomycotina appear to have a higher biosynthetic specificity toward terpenoids than other fungi, which mostly focus on polyketides production.
As explained previously, Figure @fig:magicTree is highly dependent of the data reported in literature.
As also illustrated in Figure @fig:distribution, some compounds can be over-studied among several organisms, and many organisms studied for specific compounds only.
This is a direct consequence of the way the NP community reports its data actually.
Having this in mind, when observed at a finer scale, down to the structure level, such chemotaxonomic representation can give valuable insights.
For example, among all chemical structures, only two were found in all biological kingdoms, namely heptadecanoic acid (KEMQGTRYUADPNZ-UHFFFAOYSA-N) and *β*-carotene (OENHQHLEOONYIE-JLTXGRSLSA-N).
Looking at the distribution of *β*-sitosterol (KZJWDPNRJALLNS-VJSFXXLFSA-N) within the overall biological tree, Figure @fig:magicTree-additional plots its presence/absence *versus* those of its superior chemical classifications, namely the stigmastane, steroid and terpenoid derivatives, over the same tree used in Figure @fig:magicTree.
The comparison of these five chemically-interpreted biological trees clearly highlights the increasing speciation of the *β*-sitosterol biosynthetic pathway in the Archaeplastida kingdom, while the superior classes are distributed across all kingdoms.
Figure @fig:magicTree is zoomable and vectorized for detailed inspection.

As illustrated, the possibility of data interrogation at multiple precision levels, from fully defined chemical structures to broader chemical classes, is of great interest, e.g., for taxonomic and evolution studies.
This makes LOTUS a unique resource for the advancement of chemotaxonomy, a discipline pioneered by Augustin Pyramus de Candolle and pursued by other notable researchers (Robert Hegnauer, Otto R. Gottlieb) [@doi:10.1007/978-3-642-68641-2; @doi:10/d592kt; @decandolleEssaiProprietesMedicales1816].
Six decades after Hegnauer's publication of "Die Chemotaxonomie der Pflanzen" [@doi:10.1007/978-3-0348-9314-5] much remains to be done for the advancement of this field of study and the LOTUS initiative aims to provide a solid basis for researchers willing to pursue these exciting explorations at the interface of chemistry, biology and evolution.

As shown recently in the context of spectral annotation [@doi:10.1038/s41587-020-0740-8], lowering the precision level of the annotation allows a broader coverage along with greater confidence.
Genetic studies investigating the pathways involved and the organisms carrying the responsible biosynthetic genes would be of interest to confirm the previous observations.
These forms of data interpretation exemplify the importance of reporting not only new structures, but also novel occurrences of known structures in organisms as comprehensive chemotaxonomic studies are pivotal for a better understanding of the metabolomes of living organisms.

The integration of multiple knowledge sources, e.g. genetics for NP producing gene clusters [@doi:10.1093/nar/gkz882] combined to taxonomies and occurrences DB, also opens new opportunities to understand if an organism is responsible for the *biosynthesis* of a NP or merely *contains* it.
This understanding is of utmost importance for the chemotaxonomic field and will help to understand to which extent microorganisms (endosymbionts) play a role in host development and its NP expression potential [@doi:10.1016/j.tplants.2004.04.005].


## Conclusion & Perspectives {.page_break_before}

### Advancing Natural Products Knowledge

At its current development stage, data harmonized and curated throughout the LOTUS initiative remains imperfect and, by the very nature of research, at least partially biased (see [Introduction](#introduction)).
In the context of bioactive NP research, and due to global editorial practices, it should not be ignored that many publications tend to emphasize new compounds and/or those for which interesting bioactivity has been measured.
Near-ubiquitous (primarily plant-based) compounds, if broadly bioactive, tend to be overrepresented in the NP literature, yet the implication of their wide distribution in nature and associated patterns of non-specific activity are often underappreciated [@doi:10.1021/acs.jmedchem.5b01009].
Ideally, all characterized compounds independent of structural novelty and/or bioactivity profile should be documented, and the sharing of verified structure-organism pairs is fundamental to the advancement of NP research.

The LOTUS initiative provides a framework for rigorous review and incorporation of new records and already presents a valuable overview of the distribution of NP occurrences studied to date.
While current data presents a reasonable approximation of the chemistries of a few well-studied organisms such as *Arabidopsis thaliana*, it remains patchy for many other organisms represented in the dataset.
Community participation is the most efficient means of achieving a better documentation of NP occurrences, and the comprehensive editing opportunities provided within LOTUS and through the associated Wikidata distribution platform open new opportunities for such collaborative engagement.
In addition to facilitating the introduction of new data, it also provides a forum for critical review of existing data (see an example of a Wikidata Talk page [here](https://www.wikidata.org/wiki/Talk:Q104916955)), as well as harmonization and verification of existing NP datasets as they come online.

### Fostering FAIRness and TRUSTworthiness

The LOTUS harmonized data and dissemination of referenced structure-organism pairs through Wikidata, enables novel forms of queries and transformational perspectives in NP research.
As LOTUS follows the guidelines of FAIRness and TRUSTworthiness, all researchers across disciplines can benefit from this opportunity, whether the interest is in ecology and evolution, chemical ecology, drug discovery, biosynthesis pathway elucidation, chemotaxonomy, or other research fields connected with NP.

Researchers worldwide uniformly acknowledge the limitations caused by the intrinsic unavailability of essential (raw) data [@doi:10.1039/c6np00022c].
In addition to being FAIR, LOTUS data is also open with a clear license, while closed data is still a major impediment to advancement of science [@doi:10.1038/npre.2008.1526.1].
The lack of progress in such direction is partly due to elements in the dissemination channels of the classical print and static PDF publication formats that complicate or sometimes even discourage data sharing, e.g., due to page limitations and economically motivated mechanisms, including those involved in the focus on and calculation of journal impact factors.
In particular raw data such as experimental readings, spectroscopic data, instrumental measurements, statistical, and other calculations are valued by all, but disseminated by only very few.
The immense value of raw data and the desire to advance the public dissemination has recently been documented in detail for nuclear magnetic resonance (NMR) spectroscopic data by a large consortium of NP researchers [@doi:10.1039/c7np00064b].
However, to generate the vital flow of contributed data, the effort associated with preparing and submitting content to open repositories as well as data reuse should be better acknowledged in academia, government, regulatory, and industrial environments [@doi:10/gh533h; @doi:10/gfnxpj; @doi:10/gf3j9t].
The introduction of LOTUS provides here a new opportunity to advance the FAIR guiding principles for scientific data management and stewardship [@doi:10/bdd4].

### Opening New Perspectives for Spectral Data {.page_break_before}

The possibilities for expansion and future applications of the Wikidata-based LOTUS initiative are significant.
For example, properly formatted spectral data (e.g. obtained by MS or NMR) can be linked to the Wikidata entries of the originating chemical compounds.
MassBank [@doi:10/d8pdcr] and SPLASH [@doi:10.1093/bioinformatics/btq476] identifiers are already reported in Wikidata, and this existing information can be used to report MassBank or SPLASH records for example for *Arabidopsis thaliana* compounds ([https://w.wiki/3PJD](https://w.wiki/3PJD)).
Such possibilities will help to bridge experimental data results obtained during the early stages of NP research with data that has been reported and formatted in different contexts.
This opens exciting perspectives for structural dereplication, NP annotation, and metabolomic analysis.
The authors have previously demonstrated that taxonomically-informed metabolite annotation is critical for the improvement of the NP annotation process [@doi:10/fm73].
Alternative approaches linking structural annotation to biological organisms have also shown substantial improvements [@doi:10.1101/2021.03.18.435634].
In this context, the LOTUS initiative offers new opportunities for linking chemical objects to both their biological occurrences and spectral information and should significantly facilitate such applications.

### Integrating Chemodiversity, Biodiversity, and Human Health

As shown in Figure @fig:magicTree-additional, observing the chemical and biological diversity at various granularities offers new insights.
Regarding the chemical objects involved, it will be important to document the taxonomies of chemical annotations for the Wikidata entries.
However, this is a rather complex task, for which stability and coverage issues will have to be addressed first.
Existing chemical taxonomies such as ChEBI, ClassyFire, or NPClassifier are evolving steadily, and it will be important to constantly update the tools used to make further annotations.
[Promising efforts](https://www.wikidata.org/wiki/User:SCIdude/Modeling#Chemical_ontology) have been undertaken to automate the inclusion of Wikidata structures into a chemical ontology.
Such approach exploits the [SMILES](https://www.wikidata.org/wiki/Q466769) and [SMARTS](https://www.wikidata.org/wiki/Q7544680) associated properties to infer a chemical classification for the structure.
See for example the entry related to [emericellolide B](https://www.wikidata.org/wiki/Q77573987).
Repositioning NP within their greater biosynthetic context is another major challenge - and active field of research.
The fact that the LOTUS initiative disseminates its data through Wikidata will help facilitate further integration with biological pathway knowledge bases such as [WikiPathways](https://www.wikipathways.org/index.php/WikiPathways) and contribute to this complex task [@doi:10/gh6dq2; @doi:10.1093/nar/gkx1064].

In the field of ecology, molecular traits are gaining increased attention [@doi:10/gftqcm; @doi:10/ghttk6; @doi:10.1007/s10530-018-1735-y].
The LOTUS architecture can help to associate classical plant traits (e.g., leaf surface area, photosynthetic capacities, etc.) with Wikidata biological organisms entries, and, thus, allow their integration and comparison with chemicals that are associated with the organisms.
Likewise, the association of biogeography data documented in repositories such as GBIF could be further exploited in Wikidata to pursue the exciting but understudied topic of “chemodiversity hotspots” [@doi:10/ghssx4].

Other NP-related information of great interest remains poorly formatted.
One example of such a shortcoming relates to traditional medicine (and the field studying it: ethnomedicine and ethnobotany), which is the historical and empiric approach of mankind to discover and use bioactive products from Nature, primarily plants.
The amount of knowledge generated in human history on the use of medicinal substances represents a fascinating yet underutilized amount of information.
Notably, the body of literature on the pharmacology and toxicology of NP is compound-centric, increases steadily, and relatively scattered, but still highly relevant for exploring the role and potential utility of NP for human health.
To this end, the LOTUS initiative represents a potential framework for new concepts by which such information could be valued and conserved in the digital era [@doi:10/ggb79p; @doi:10/gh533j; @doi:10.1177/1934578X1701200849].
This underscores the transformative value of the LOTUS initiative for the advancement of traditional medicine and its interest for drug discovery in health systems worldwide.

### Shortcomings & Challenges {.page_break_before}

Despite these strong advantages, the establishment and functioning of the LOTUS curation pipeline is not devoid of defaults and we list hereafter some of the observed shortcomings and associated challenges.

First, the LOTUS processing pipeline is heavy.
It includes many dependencies and is convoluted.
We tried to simplify the process and associated programs as much as possible but they remain consequent.
This is the consequence of the heterogeneous nature of the source information and the number of successive operations required to process the data.

Second, while the overall objective of the LOTUS processing pipeline is to increase data quality, the pipeline also _transforms_ data during the process, and, in some cases, data quality can be degraded or errors can be propagated.
For example, regarding the chemical objects, the processing pipeline performs a systematic sanitization step that includes salt removal, uncharging of molecules and dimers resolving.
We decided to apply this step systematically after observing a high ratio of artifacts within salts, charged or dimeric molecules.
This thus implies that correct salts, charged or dimeric molecules in the input data will suffer an unwanted "sanitization" step.
Also, the LOTUS processing step uses external libraries and tools for the automated "name to structure" and "structure to name" translations.
These remain challenging as they rely on sets of predefined rules which do not cover all cases and can commonly lead to incorrect translations.

On the biological organisms curation side, we are aware of shortcomings, whether inherent to specific inputs or regarding limitations of the general process.
Regarding inputs, some cases are clearly not resolvable except through human curation.
For example, the word Lotus can refer both to the genus of a plant of the Fabaceae family ([https://www.wikidata.org/wiki/Q3645698](https://www.wikidata.org/wiki/Q3645698)) or to the vernacular name of _Nelumbo nucifera_ ([https://www.wikidata.org/wiki/Q16528)](https://www.wikidata.org/wiki/Q16528))).
In fact, the name of the LOTUS Initiative comes, in part, from this taxonomic curiosity - and the challenge for its automated curation.
To give another striking illustration, _Ficus variegata_ corresponds both to a plant ([https://www.wikidata.org/wiki/Q5446649](https://www.wikidata.org/wiki/Q5446649)) and to a mollusc ([https://www.wikidata.org/wiki/Q502030](https://www.wikidata.org/wiki/Q502030)).
For specific names coming from traditional Chinese medicine or other sources using vernacular names, translation was dependent on hand curated dictionaries, which are clearly not exhaustive.
Additionally, it is noteworthy to remind that the validation of the processed entries relies in part on partly imperfect rules, thus leading to erroneous entries in the output data.
However, we also deliberately kept those rules restrictive in order to overall favor quality over quantity (see Figure @fig:alluvial).

Thus, despite our efforts, there is no doubt that incorrect structure-organism pairs have been uploaded on Wikidata (and some correct ones have not).
We however expect that the editing facilities offered by this platform and community efforts will, over time, improve data quality.


### Summary & Outlook

Despite these challenges, the various facets discussed above connect with ongoing and future developments that the tandem of the LOTUS initiative and its Wikidata integration can accommodate through a broader knowledge base.
The information of the LOTUS initiative is already readily accessible by third party projects build on top of Wikidata such as the SLING project ([https://github.com/ringgaard/sling](https://github.com/ringgaard/sling), see entry for [gliotoxin](https://ringgaard.com/kb/Q413364)) or the Plant Humanities Lab project ([https://lab.plant-humanities.org](https://lab.plant-humanities.org), see entry for [*Ilex guayusa*](https://search.plant-humanities.org/?eid=wd%3AQ3772280) in the "From Related Items" section).
LOTUS data has also been integrated to PubChem ([https://pubchem.ncbi.nlm.nih.gov/source/25132](https://pubchem.ncbi.nlm.nih.gov/source/25132)) to complement the natural products related metadata of this major chemical DB.
For an example, see [*Gentiana lutea*](https://pubchem.ncbi.nlm.nih.gov/taxonomy/38851#section=Natural-Products).

Behind the scenes, all underlying resources represent data in a multidimensional space and can be extracted as individual graphs, which can then be interconnected.
The craft of appropriate federated queries allows users to navigate these graphs and fully exploit their potential [@doi:10/gdn52j; @doi:10/ggqqc6].
The development of interfaces such as RDFFrames [@wikidata:Q106204599] will also facilitate the use of the wide arsenal of existing machine learning approaches to automate reasoning on these knowledge graphs.

Overall, the LOTUS initiative aims to make more and better data available.
While we did our best to ensure high data quality, the current processing pipeline still removes a lot of correct entries and misses or induces some incorrect ones.
Aware of those imperfections, our project hopefully paves the way for the establishment of an open, durable and expandable electronic NP resource.
The design and efforts of the LOTUS initiative reflect our conviction that the integration of NP research results is long-needed and requires a truly open and FAIR approach to information dissemination, with high quality data directly flowing from its source to public knowledge bases.
We believe that the LOTUS initiative has the potential to fuel a virtuous cycle of research habits and, as a result, *contribute to a better understanding of Life and its chemistry*.


## Methods {.page_break_before}

### Key Resources

<div style="font-size:9pt">

Table: Key resources {#tbl:key-resources}

| Reagent type (species) or resource | Designation | Source or reference | Identifiers | Additional information | 
|:---:|:---:|:------:|:---:|:------:| 
| Software, algorithm| Lotus-processor code| This work ([https://github.com/lotusnprod/lotus-processor](https://github.com/lotusnprod/lotus-processor))|| Archived at [https://doi.org/10.5281/zenodo.5802107](https://doi.org/10.5281/zenodo.5802107) | 
| Software, algorithm| Lotus-web code| This work ([https://github.com/lotusnprod/lotus-web](https://github.com/lotusnprod/lotus-web))|| Archived at [https://doi.org/10.5281/zenodo.5802119](https://doi.org/10.5281/zenodo.5802119) | 
| Software, algorithm| Lotus-wikidata-interact code| This work ([https://github.com/lotusnprod/lotus-wikidata-interact](https://github.com/lotusnprod/lotus-wikidata-interact)) || Archived at [https://doi.org/10.5281/zenodo.5802113](https://doi.org/10.5281/zenodo.5802113) | 
| Software, algorithm| Global Names Architeture| [https://globalnames.org](https://globalnames.org) | QID:[Q65691453](https://www.wikidata.org/wiki/Q65691453)| See [Additional executable files](#additional-executable-files)  | 
| Software, algorithm| Java| [https://www.java.com](https://www.java.com) | QID:[Q251](https://www.wikidata.org/wiki/Q251)| | 
| Software, algorithm| Kotlin| [https://kotlinlang.org](https://kotlinlang.org) | QID:[Q3816639](https://www.wikidata.org/wiki/Q3816639)| See [Kotlin packages](#kotlin) | 
| Software, algorithm| Manubot| [https://manubot.org](https://manubot.org) | QID:[Q96473455](https://www.wikidata.org/wiki/Q96473455) RRID:SCR_018553 | Repository available at [https://github.com/lotusnprod/lotus-manuscript](https://github.com/lotusnprod/lotus-manuscript) | 
| Software, algorithm| NPClassifier| [https://npclassifier.ucsd.edu](https://npclassifier.ucsd.edu) || See [https://doi.org/10.1021/acs.jnatprod.1c00399](https://doi.org/10.1021/acs.jnatprod.1c00399) | 
| Software, algorithm| OPSIN| [https://github.com/dan2097/opsin](https://github.com/dan2097/opsin) | QID:[Q26481302](https://www.wikidata.org/wiki/Q26481302)| See [Additional executable files](#additional-executable-files) | 
| Software, algorithm| Python Programming Language| [https://www.python.org](https://www.python.org) | QID:[Q28865](https://www.wikidata.org/wiki/Q28865) RRID:SCR_008394| See [Python packages](#python) | 
| Software, algorithm| R Project for Statistical Computing| [https://www.r-project.org](https://www.r-project.org) | QID:[Q206904](https://www.wikidata.org/wiki/Q206904) RRID:SCR_001905| See [R packages](#r) | 
| Software, algorithm| Molconvert| [https://docs.chemaxon.com/display/docs/molconvert.md](https://docs.chemaxon.com/display/docs/molconvert.md) | QID:[Q55377678](https://www.wikidata.org/wiki/Q55377678)| See [Chemical structures](#chemical-structures) | 
| Software, algorithm| Wikidata | [https://www.wikidata.org](https://www.wikidata.org) | QID:[Q2013](https://www.wikidata.org/wiki/Q2013) RRID:SCR_018492| Project page [https://www.wikidata.org/wiki/Wikidata:WikiProject_Chemistry/Natural_products](https://www.wikidata.org/wiki/Wikidata:WikiProject_Chemistry/Natural_products)|
| Other| Lotus custom dictionaries| This work || Archived at [https://doi.org/10.5281/zenodo.5801798](https://doi.org/10.5281/zenodo.5801798) | 
| Other| Chemical identifier resolver| [https://cactus.nci.nih.gov/chemical/structure](https://cactus.nci.nih.gov/chemical/structure) || See [Chemical structures](#chemical-structures)| 
| Other| CrossRef | [https://www.crossref.org](https://www.crossref.org) | QID:[Q5188229](https://www.wikidata.org/wiki/Q5188229) RRID:SCR_003217|See [References](#references)|
| Other| PubChem| [https://pubchem.ncbi.nlm.nih.gov](https://pubchem.ncbi.nlm.nih.gov) | QID:[Q278487](https://www.wikidata.org/wiki/Q278487) RRID:SCR_004284| LOTUS data [https://pubchem.ncbi.nlm.nih.gov/source/25132](https://pubchem.ncbi.nlm.nih.gov/source/25132) | 
| Other| PubMed| [https://pubmed.ncbi.nlm.nih.gov](https://pubmed.ncbi.nlm.nih.gov) | QID:[Q180686](https://www.wikidata.org/wiki/Q180686) RRID:SCR_004846|See [References](#references)|
| Other| Taxonomic data sources | [https://resolver.globalnames.org/data_sources](https://resolver.globalnames.org/data_sources) || See [Translation](#translation) | 
| Other| Natural Products data sources ||| See [Appendix 1](#appendix-1-data-sources-list) | 

</div>

### Data Gathering {.page_break_before}

Before their inclusion, the overall quality of the source was manually assessed to estimate, both, the quality of referenced structure-organism pairs and the lack of ambiguities in the links between data and references.
This led to the identification of thirty-six electronic NP resources as valuable LOTUS input.
Data from the proprietary Dictionary of Natural Products (DNP v 29.2) was also used for comparison purposes only and is not publicly disseminated.
[FooDB](https://foodb.ca) was also curated but not publicly disseminated since its license proscribed sharing in Wikidata.
[Appendix 1](#appendix-1-data-sources-list) gives all necessary details regarding electronic NP resources access and characteristics.

Manual inspection of each electronic NP resource revealed that the structure, organism, and reference fields were widely variable in format and contents, thus requiring standardization to be comparable.
The initial stage consisted of writing tailored scripts that are capable of harmonizing and categorizing knowledge from each source (Figure @fig:workflow).
This transformative process led to three categories: fields relevant to the chemical structure described, to the producing biological organism, and the reference describing the occurrence of the chemical structure in the producing biological organism.
This process resulted in categorized columns for each source, providing an initial harmonized format for each table.

For all thirty-eight sources, if a single file or multiple files were accessible *via* a download option including FTP, data was gathered that way.
For some sources, data was scraped (cf. [Appendix 1](#appendix-1-data-sources-list)).
All scraping scripts can be found in the [lotus-processor](https://github.com/lotusnprod/lotus-processor) repository in the [src/1_gathering](https://github.com/lotusnprod/lotus-processor/tree/main/src/1_gathering) directory (under each respective subdirectory).
Data extraction scripts for the DNP are available and should allow users with a DNP license only to further exploit the data ([src/1_gathering/db/dnp](https://github.com/lotusnprod/lotus-processor/tree/main/src/1_gathering/db/dnp)).
The chemical structure fields, organism fields, and reference fields were manually categorized into three, two, and ten subcategories, respectively.
For chemical structures, "InChI", "SMILES", and "chemical name" (not necessarily IUPAC).
For organisms, "clean" and "dirty", meaning lot text not referred to the canonical name was present or the organism was not described by its canonical name (e.g. "Compound isolated from the fresh leaves of *Citrus* spp.").
For the references, the original reference was kept in the "original" field.
When the format allowed it, references were divided into: "authors", "doi", "external", "isbn", "journal", "original", "publishing details", "pubmed", "title", "split".
The generic "external" field was used for all external cross-references to other websites or electronic NP resources (e.g. "also in knapsack").
The last subcategory, "split", corresponds to a still non-atomic field after the removal of parts of the original reference.
Other field titles are self-explanatory.
The producing organism field was kept as a single field.

### Data Harmonization

To perform the harmonization of all previously gathered sources, sixteen columns were chosen as described above.
Upon electronic NP resources harmonization, resulting subcategories were divided and subject to further processing.
The "chemical structure" fields were divided into files according to their subcategories ("InChI", "names" and "SMILES").
A file containing all initial structures from all three subcategories was also generated.
The same procedure was followed for organisms and references.

### Data Processing

To obtain an unambiguously referenced structure-organism pair for Wikidata dissemination, the initial sixteen columns were translated and processed into three fields: the reported structure, the organism canonical name, and the reference.
The structure was reported as InChI, together with its SMILES and InChIKey translation.
The biological organism field was reported as three minimal necessary and sufficient fields, namely its canonical name and the taxonID and taxonomic DB corresponding to the latter.
The reference was reported as four minimal fields, namely reference title, DOI, PMCID, and PMID, one being sufficient.
For the forthcoming translation processes, automated solutions were used when available.
However, for specific cases (common or vernacular names of the biological organisms, Traditional Chinese Medicine (TCM) names, and conversion between digital reference identifiers), no solution existed, thus requiring the use of tailored dictionaries.
Their construction is detailed in the [Dictionaries](#dictionaries) section.
The initial entries (containing one or multiple producing organisms per structure, with one or multiple accepted names per organism) were processed into 2M+ referenced structure-organism pairs.

#### Chemical Structures

To retrieve as much information as possible from the original structure field(s) of each of the sources, the following procedure was followed.
Allowed structural fields for the sources were divided into two types: structural (InChI, SMILES) or nominal (chemical name, not necessarily IUPAC).
If multiple fields were present, structural identifiers were preferred over structure names.
Among structural identifiers, when both identifiers were present, SMILES was preferred over InChI.
InChI were translated to SMILES using the RDKit (2021.09.2) implementation in Python 3.8 ([src/2_curating/2_editing/structure/1_translating/inchi.py](https://github.com/lotusnprod/lotus-processor/tree/main/src/2_curating/2_editing/structure/1_translating/inchi.py)).
They were first converted to [ROMol](https://www.rdkit.org/docs/cppapi/ROMol_8h.html#details) objects which were then converted to SMILES.
When no structural identifier was available, the nominal identifier was translated to InChI first thanks to [OPSIN](https://github.com/dan2097/opsin) [@doi:10/b2zkr9], a fast Java-based translation open-source solution.
If no translation was obtained, chemical names were then submitted to the PUG-REST, the interface for programmatic access to PubChem [@doi:10.1093/nar/gky294; @doi:10.1093/nar/gkv396].
If again no translation was obtained, candidates were then submitted to the [Chemical Identifier Resolver](https://cactus.nci.nih.gov).
Before the translation process, some typical chemical structure-related greek characters (such as *α*, *ß*) were replaced by their textual equivalents (alpha, beta) to obtain better results.
All pre-translation steps are included in the preparing_name function and are available in [src/r/preparing_name.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/r/preparing_name.R).

The chemical sanitization step sought to standardize the representation of chemical structures coming from different sources.
It consisted of three main stages (standardizing, fragment removal, and uncharging) achieved *via* the MolVS package.
The initial standardizer function consists of six stages (RDKit Sanitization, RDKit Hs removal, Metals Disconnection, Normalization, Acids Reionization, and Stereochemistry recalculation) detailed in the [molvs documentation](https://molvs.readthedocs.io/en/latest/guide/standardize.html).
In a second step, the FragmentRemover functionality was applied using a list of SMARTS to detect and remove common counterions and crystallization reagents sometimes occurring in the input DB.
Finally, the Uncharger function was employed to neutralize molecules when appropriate.

[Molconvert](https://docs.chemaxon.com/display/docs/molconvert.md) function of the MarvinSuite was used for traditional and IUPAC names translation, Marvin 20.19, [ChemAxon](https://www.chemaxon.com).
When stereochemistry was not fully defined, (+) and (-) symbols were removed from names.
All details are available in the following script: [src/2_curating/2_editing/structure/4_enriching/naming.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/2_curating/2_editing/structure/4_enriching/naming.R).
Chemical classification of all resulting structures was done using classyfireR [@doi:10/gc5tqv] and [NPClassifier API](https://ccms-ucsd.github.io/GNPSDocumentation/api/#structure-np-classifier).

After manual evaluation, structures remaining as dimers were discarded (all structures containing a "." in their SMILES were removed).

From the 283,267 initial InChI, 242,068 (85%) sanitized structures were obtained, of which 185,929 (77%) had complete stereochemistry defined.
203,718 (72%) were uploaded to Wikidata.
From the 248,185 initial SMILES, 207,658 (84%) sanitized structures were obtained, of which 98,685 (48%) had complete stereochemistry defined.
174,091 (70%) were uploaded to Wikidata.
From the 49,675 initial chemical names, 27,932 (56%) sanitized structures were obtained, of which 17,460 (63%) had complete stereochemistry defined.
23,036 (46%) were uploaded to Wikidata.
In total, 163,800 structures with fully defined stereochemistry were uploaded as "chemical compounds" ([Q11173](https://www.wikidata.org/wiki/Q11173)), and 106,669 structures without fully defined stereochemistry were uploaded as "group of stereoisomers" ([Q59199015](https://www.wikidata.org/wiki/Q59199015)).


#### Biological Organisms {.page_break_before}

The processing at the biological organism’s level had three objectives: convert the original organism string to (a) taxon name(s), atomize fields containing multiple taxon names, and deduplicate synonyms.
The original organism strings were treated with [Global Names Finder](https://github.com/gnames/gnfinder) (GNF) and [Global Names Verifier](https://github.com/gnames/gnverifier) (GNV), both tools coming from the [Global Names Architecture](http://globalnames.org) (GNA) a system of web services that helps people to register, find, index, check and organize biological scientific names and interconnect on-line information about species.
GNF allows scientific name recognition within raw text blocks and searches for found scientific names among public taxonomic DB.
GNV takes names or lists of names and verifies them against various biodiversity data sources.
Canonical names, their taxonID, and the taxonomic DB they were found in were retrieved.
When a single entry led to multiple canonical names (accepted synonyms), all of them were kept.
Because both GNF and GNV recognize scientific names and not common ones, common names were translated before a second resubmission.

##### Dictionaries

To perform the translations from common biological organism name to latin scientific name, specialized dictionaries included in DrDuke, FooDB, PhenolExplorer were aggregated together with the translation dictionary of [GBIF Backbone Taxonomy](https://www.gbif.org/dataset/d7dddbf4-2cf0-4f39-9b2a-bb099caae36c).
The script used for this was [src/1_gathering/translation/common.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/1_gathering/translation/common.R).
When the canonical translation of a common name contained a specific epithet that was not initially present, the translation pair was discarded (for example, "Aloe" translated in "*Aloe vera*" was discarded).
Common names corresponding to a generic name were also discarded (for example "Kiwi" corresponding to the synonym of an *Apteryx* spp. ([https://www.gbif.org/species/4849989](https://www.gbif.org/species/4849989))).
When multiple translations were given for a single common name, the following procedure was followed: the canonical name was split into species name, genus name, and possible subnames.
For each common name, genus names and species names were counted.
If both the species and genus names were consistent at more than 50%, they were considered consistent overall and, therefore, kept (for example, "Aberrant Bush Warbler" had "*Horornis flavolivaceus*" and "*Horornis flavolivaceus intricatus*" as translation; as both the generic ("*Horornis*") and the specific ("*flavolivaceus*") epithets were consistent at 100%, both ("*Horornis flavolivaceus*") were kept).
When only the generic epithet had more than 50% consistency, it was kept (for example, "Angelshark" had "*Squatina australis*" and "*Squatina squatina*" as translation, so only "*Squatina*" was kept).
Some unspecific common names were removed (see [https://doi.org/10.5281/zenodo.5801816](https://doi.org/10.5281/zenodo.5801816) [@doi:10.5281/zenodo.5801816]) and only common names with more than three characters were kept.
This resulted in 181,891 translation pairs further used for the conversion from common names to scientific names.
For TCM names, translation dictionaries from TCMID, TMMC, and coming from the Chinese Medicine Board of Australia were aggregated.
The script used for this was [src/1_gathering/translation/tcm.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/1_gathering/translation/tcm.R).
Some unspecific common names were removed (see [https://doi.org/10.5281/zenodo.5801816](https://doi.org/10.5281/zenodo.5801816) [@doi:10.5281/zenodo.5801816]).
Careful attention was given to the Latin genitive translations and custom dictionaries were written (see [https://doi.org/10.5281/zenodo.5801816](https://doi.org/10.5281/zenodo.5801816) [@doi:10.5281/zenodo.5801816]).
Organ names of the producing organism were removed to avoid wrong translation (see [https://doi.org/10.5281/zenodo.5801816](https://doi.org/10.5281/zenodo.5801816) [@doi:10.5281/zenodo.5801816]).
This resulted in 7,070 translation pairs.
Both common and TCM translation pairs were then ordered by decreasing string length, first translating the longer names to avoid part of them being translated incorrectly.

##### Translation

To ensure compatibility between obtained taxonID with Wikidata, the taxonomic DB 3 ([ITIS](https://www.itis.gov)), 4 ([NCBI](https://www.ncbi.nlm.nih.gov/taxonomy)), 5 ([Index Fungorum](http://www.indexfungorum.org)), 6 ([GRIN Taxonomy for Plants](http://wgb.cimmyt.org/gringlobal/taxon/abouttaxonomy.aspx?language=en&chapter=scope)), 8 ([The Interim Register of Marine and Nonmarine Genera](https://www.irmng.org)), 9 ([World Register of Marine Species](http://www.marinespecies.org)), 11 ([GBIF Backbone Taxonomy](https://www.gbif.org)), 12 ([Encyclopedia of Life](https://eol.org)), 118 ([AmphibiaWeb](https://amphibiaweb.org)), 128 ([ARKive](https://www.wildscreen.org/arkive-closure)), 132 ([ZooBank](http://zoobank.org)), 147 ([Database of Vascular Plants of Canada (VASCAN)](http://data.canadensys.net/vascan/search)), 148 ([Phasmida Species File](http://phasmida.speciesfile.org)), 150 ([USDA NRCS PLANTS Database](https://plants.sc.egov.usda.gov)), 155 ([FishBase](https://www.fishbase.se)), 158 ([EUNIS](https://eunis.eea.europa.eu)), 163 ([IUCN Red List of Threatened Species](https://www.iucnredlist.org)), 164 ([BioLib.cz](https://www.biolib.cz)), 165 ([Tropicos - Missouri Botanical Garden](https://www.tropicos.org)), 167 ([The International Plant Names Index](https://www.ipni.org)), 169 ([uBio NameBank](http://ubio.org)), 174 ([The Mammal Species of The World](https://www.departments.bucknell.edu/biology/resources/msw3)), 175 ([BirdLife International](http://www.birdlife.org)), 179 ([Open Tree of Life](https://tree.opentreeoflife.org)), 180 ([iNaturalist](https://www.inaturalist.org)) and 187 ([The eBird/Clements Checklist of Birds of the World](https://ebird.org/science/use-ebird-data/the-ebird-taxonomy)) were chosen.
All other available taxonomic DB are listed at [http://index.globalnames.org/datasource](http://index.globalnames.org/datasource).
To retrieve as much information as possible from the original organism field of each of the sources, the following procedure was followed: First, a scientific name recognition step, allowing us to retrieve canonical names was carried ([src/2_curating/2_editing/organisms/subscripts/1_processingOriginal.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/2_curating/2_editing/organisms/subscripts/1_processingOriginal.R)).
Then, a subtraction step of the obtained canonical names from the original field was applied, to avoid unwanted translation of parts of canonical names.
For example, _Bromus mango_ contains "mango" as a specific epithet, which is also the common name for _Mangifera indica_.
After this subtraction step, the remaining names were translated from vernacular (common) and TCM names to scientific names, with help of the dictionaries.
For performance reasons, this processing step was written in Kotlin and used coroutines to allow efficient parallelization of that process ([src/2_curating/2_editing/organisms/2_translating_organism_kotlin/](https://github.com/lotusnprod/lotus-processor/tree/main/src/2_curating/2_editing/organisms/2_translating_organism_kotlin/)).
They were subsequently submitted again to scientific name recognition ([src/2_curating/2_editing/organisms/3_processingTranslated.R](http://src/2_curating/2_editing/organisms/3_processingTranslated.R)).

After full resolution of canonical names, all obtained names were submitted to rotl [@doi:10.1111/2041-210x.12593] to obtain a unified taxonomy.
From the 88,395 initial "clean" organism fields, 43,936 (50%) canonical names were obtained, of which 32,285 (37%) were uploaded to Wikidata.
From the 300 initial "dirty" organism fields, 250 (83%) canonical names were obtained, of which 208 (69%) were uploaded to Wikidata.

#### References 

The [Rcrossref](https://cran.r-project.org/web/packages/rcrossref) package [@chamberlainRcrossrefClientVarious2020] interfacing with the [Crossref](https://www.crossref.org) API was used to translate references from their original subcategory ("original", "publishingDetails", "split", "title") to a DOI, the title of its corresponding article, the journal it was published in, its date of publication and the name of the first author.
The first twenty candidates were kept and ranked according to the score returned by Crossref, which is a [tf-idf](https://lucene.apache.org/core/8_8_0/core/org/apache/lucene/search/similarities/TFIDFSimilarity.html) score.
For DOI and PMID, only a single candidate was kept.
All DOIs were also translated with this method, to eventually discard any DOI not leading to an object.
PMIDs were translated, thanks to the entrez_summary function of the [rentrez](https://cran.r-project.org/web/packages/rentrez) package [@doi:10/gh5ptg].
Scripts used for all subcategories of references are available in the directory [src/2_curating/2_editing/reference/1_translating/](https://github.com/lotusnprod/lotus-processor/tree/main/src/2_curating/2_editing/reference/1_translating/).
Once all translations were made, results coming from each subcategory were integrated, ([src/2_curating/2_editing/reference/2_integrating.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/2_curating/2_editing/reference/2_integrating.R)) and the producing organism related to the reference was added for further treatment.
Because the crossref score was not informative enough, at least one other metric was chosen to complement it.
The first metric was related to the presence of the producing organism’s generic name in the title of the returned article.
If the title contained the generic name of the organism, a score of 1 was given, else 0.
Regarding the subcategories "doi", "pubmed" and "title", for which the same subcategory was retrieved *via* crossref or rentrez, distances between the input’s string and the candidates’ one were calculated.
Optimal string alignment (restricted [Damerau-Levenshtein distance](https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance)) was used as a method.
Among "publishing details", "original" and "split" categories, three additional metrics were used: If the journal name was present in the original field, a score of 1 was given, else 0.
If the name of the first author was present in the original field, a score of 1 was given, else 0.
Those three scores were then summed together.
All candidates were first ordered according to their crossref score, then by the complement score for related subcategories, then again according to their title-producing organism score, and finally according to their translation distance score.
After this re-ranking step, only the first candidate was kept.
Finally, the Pubmed PMCID dictionary ([PMC-ids.csv.gz](https://ftp.ncbi.nlm.nih.gov/pub/pmc/PMC-ids.csv.gz)) was used to perform the translations between DOI, PMID, and PMCID ([src/2_curating/2_editing/reference/3_processing.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/2_curating/2_editing/reference/3_processing.R)).

From the 36,710 initial "original" references, 21,970 (60%) references with sufficient quality were obtained, of which 15,588 (71%) had the organism name in their title.
14,710 (40%) were uploaded to Wikidata.
From the 21,953 initial "pubmed" references, 9,452 (43%) references with sufficient quality were obtained, of which 6,098 (65%) had the organism name in their title.
5,553 (25%) were uploaded to Wikidata.
From the 37,371 initial "doi" references, 20,139 (54%) references with sufficient quality were obtained, of which 15,727 (78%) had the organism name in their title.
15,351 (41%) were uploaded to Wikidata.
From the 29,600 initial "title" references, 17,417 (59%) references with sufficient quality were obtained, of which 12,675 (73%) had the organism name in their title.
10,725 (36%) were uploaded to Wikidata.
From the 11,325 initial "split" references, 5,856 (52%) references with sufficient quality were obtained, of which 3,206 (55%) had the organism name in their title.
2,854 (25%) were uploaded to Wikidata.
From the 3,314 initial "publishingDetails" references, 119 (4%) references with sufficient quality were obtained, of which 59 (50%) had the organism name in their title.
58 (2%) were uploaded to Wikidata.

### Data Realignment

In order to fetch back the referenced structure-organism pairs links in the original data, the processed structures, processed organisms, and processed references were re-aligned with the initial entries.
This resulted in 6.2M+ referenced structure-organism pairs.
Those pairs were not unique, with redundancies among electronic NP resources and different original categories leading to the same final pair (for example, entry reporting InChI=1/C21H20O12/c22-6-13-15(27)17(29)18(30)21(32-13)33-20-16(28)14-11(26)4-8(23)5-12(14)31-19(20)7-1-2-9(24)10(25)3-7/h1-5,13,15,17-18,21-27,29-30H,6H2/t13-,15+,17+,18-,21+/m1/s1 in *Crataegus oxyacantha* or InChI=1S/C21H20O12/c22-6-13-15(27)17(29)18(30)21(32-13)33-20-16(28)14-11(26)4-8(23)5-12(14)31-19(20)7-1-2-9(24)10(25)3-7/h1-5,13,15,17-18,21-27,29-30H,6H2/t13-,15+,17+,18-,21+/m1/s1 in *Crataegus stevenii* both led to OVSQVDMCBVZWGM-DTGCRPNFSA-N in *Crataegus monogyna*).
After deduplication, 2M+ unique structure-organism pairs were obtained.

After the curation of all three objects, all of them were put together again.
Therefore, the original aligned table containing the original pairs was joined with each curation result.
Only entries containing a structure, an organism, and a reference after curation were kept.
Each curated object was divided into minimal data (for Wikidata upload) and metadata.
A dictionary containing original and curated object translations was written for each object to avoid those translations being made again during the next curation step ([src/2_curating/3_integrating.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/2_curating/3_integrating.R)).

### Data Validation 

The pairs obtained after curation were of different quality.
Globally, structure and organism translation was satisfactory whereas reference translation was not.
Therefore, to assess the validity of the obtained results, a randomized set of 420 referenced structure-organism pairs was sampled in each reference subcategory and validated or rejected manually.
Entries were sampled with at least 55 of each reference subcategory present (to get a representative idea of each subcategory) ([src/3_analyzing/1_sampling.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/3_analyzing/1_sampling.R)).
An entry was only validated if: *i)* the structure (as any structural descriptor that could be linked to the final sanitized InChIKey) was described in the reference *ii)* the producing organism (as any organism descriptor that could be linked to the accepted canonical name) was described in the reference and *iii)* the reference was describing the occurrence of the chemical structure in the biological organism.
Results obtained on the manually analyzed set were categorized according to the initial reference subcategory and are detailed in [Appendix 2](#appendix-2-summary-of-the-validation-statistics).
To improve these results, further processing of the references was needed.
This was done by accepting entries whose reference was coming from a DOI, a PMID, or from a title which restricted Damerau-Levenshtein distance between original and translated was lower than ten or if it was coming from one of the three main journals where NP occurrences are commonly expected to be published (i.e., *Journal of Natural Products*, *Phytochemistry*, or *Journal of Agricultural and Food Chemistry*).
For "split", "publishingDetails" and "original" subcategories, the year of publication of the obtained reference, its journal, and the name of the first author were searched in the original entry and if at least two of them were present, the entry was kept.
Entries were then further filtered to keep the ones where the reference title contained the first element of the detected canonical name.
Except for COCONUT, exceptions to this filter were made for all DOI-based references.
The function resulting from those rules is ([filter_dirty.R](https://github.com/lotusnprod/lotus-processor/blob/main/src/r/filter_dirty.R)).
To validate those filtering criteria, an additional set of 100 structure-organism pairs were manually analyzed.
F0.5 score was used as a metric.
F0.5 score is a modified F1 score where precision has twice more weight than recall.
The F-score was calculated with ß = 0.5, as in Equation @eq:validation: 

${F_{\beta}\ =\ \left(1+\beta^2\right)\cdot\frac{precision\cdot recall}{\left(\beta^2\ \cdot\ precision\right)\ +\ recall}}$ {#eq:validation}

Based on this first manually validated dataset, filtering criteria ([src/r/filter_dirty.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/r/filter_dirty.R)) were established to maximize precision and recall.
Another 100 entries were sampled, this time respecting the whole set ratios.
After manual validation, 97% of true positives were reached on the second set.
A summary of the validation results is given in [Appendix 2](#appendix-2-summary-of-the-validation-statistics).
Once validated, the filtering criteria were established to the whole curated set to filter entries chosen for dissemination ([src/3_analyzing/2_validating.R](https://github.com/lotusnprod/lotus-processor/tree/main/src/3_analyzing/2_validating.R)).

### Unit Testing

To provide robustness of the whole process and code, unit tests and partial data full-tests were written.
They can run on the developer machine but also on the CI/CD system (GitHub) upon each commit to the codebase.

Those tests assess that the functions are providing results coherent with what is expected especially for edge cases detected during the development.
The Kotlin code has tests based on JUnit and code quality control checks based on Ktlint, Detekt and Ben Mane's version plugin.

### Data Dissemination

#### Wikidata

All the data produced for this work has been made available on Wikidata under a [Creative Commons 0](https://creativecommons.org/publicdomain/zero/1.0) license according to [Wikidata:Licensing](https://www.wikidata.org/wiki/Wikidata:Licensing).
This is a "No-rights-reserved" license that places no restrictions on reuse.

#### Lotus.NaturalProducts.Net (LNPN)

The web interface is implemented following the same protocol as described in the COCONUT publication [@doi:10/ghssx5] i.e. the data are stored in a MongoDB repository, the backend runs with Kotlin and Java, using the Spring framework, and the frontend is written in React.js, and completely Dockerized.
In addition to the diverse search functions available through this web interface, an API is also implemented, allowing programmatic LNPN querying.
The complete API usage is described on the "Documentation" page of the website.
LNPN is part of the NaturalProducts.net portal, an initiative aimed at gathering diverse open NP resources in one place.

### Data Interaction

#### Data Retrieval

Bulk retrieval of a frozen (2021-12-20) version of LOTUS data is also available at [https://doi.org/10.5281/zenodo.5794106](https://doi.org/10.5281/zenodo.5794106) [@doi:10.5281/zenodo.5794106].


The [download lotus](https://github.com/lotusnprod/lotus-wikidata-interact/blob/main/downloadLotus/README.md) part of [lotus-wikidata-interact](https://github.com/lotusnprod/lotus-wikidata-interact) allows the download of all chemical compounds with a "found in taxon" property.
That way, it does not only get the data produced by this work, but any that would have existed beforehand or that would have been added directly on Wikidata by our users.
It makes a copy of all the entities (compounds, taxa, references) into a local triplestore that can be queried with SPARQL as is or converted to a TSV file for inclusion in other projects.
It is currently adapted to export directly into the SSOT thus allowing direct reuse by the processing/curation pipeline.

#### Data Addition 

##### Wikidata

Data is loaded by the Kotlin importer available in the [upload lotus](https://github.com/lotusnprod/lotus-wikidata-interact/blob/main/uploadLotus/README.md) part of [lotus-wikidata-interact](https://github.com/lotusnprod/lotus-wikidata-interact) repository under a GPL V3 license and imported into Wikidata.
The importer processes the curated outputs grouping references, organisms, and compounds together.
It then checks if they already exist in Wikidata (using SPARQL or a direct connection to Wikidata depending on the kind of data).
It then uses *update* or *insert*, also called *upsert*, the entities as needed.
The script currently takes the tabular file of the referenced structure-organism pairs resulting from the LOTUS curation process as input.
Before upload, a filtering step is performed in order to avoid re-uploading entries we already uploaded.
This way, if modifications occur in Wikidata, it will not be erased by the next iteration of the importer.
The importer is currently being adapted to use directly the SSOT and avoid an unnecessary conversion step.
To import references, it first double checks for the presence of duplicated DOIs and utilizes the [Crossref REST API](https://www.crossref.org/education/retrieve-metadata/rest-api) to retrieve metadata associated with the DOI, the support for other citation sources such as Europe PMC is in progress.
The structure-related fields are only subject to limited processing: basic formatting of the molecular formula by subscripting of the numbers.
Due to limitations in Wikidata, the molecule names are dropped if they are longer than 250 characters and likewise the InChI strings cannot be stored if they are longer than 1500 characters.

Uploaded taxonomical DB identifiers are currently restricted to ITIS, GBIF, NCBI Taxon, Index Fungorum, IRMNG, WORMS, VASCAN, and iNaturalist, and newly OTL.
The taxa levels are currently limited to family, subfamily, tribe, subtribe, genus, species, variety.
The importer checks for the existence of each item based on their InChIKey and upserts the compound with the *found in taxon* statement and the associated organisms and references.

##### LNPN

From the onset, LNPN has been importing data directly from the frozen tabular data of the LOTUS dataset ([https://doi.org/10.5281/zenodo.5794106](https://doi.org/10.5281/zenodo.5794106) [@doi:10.5281/zenodo.5794106]).
In future versions, LNPN will directly feed on the SSOT.

##### Data Edition

The bot framework [lotus-wikidata-interact](https://github.com/lotusnprod/lotus-wikidata-interact) was adapted such that, in addition to batch upload capabilities, it can also edit erroneously created entries on Wikidata.
As massive edits have a large potential to disrupt otherwise good data, progressive deployment of this script is used, starting by editing progressively 1, 10, then 100 entries that are manually checked.
Upon validation of 100 entries, the full script is run and check its behavior checked at regular intervals.
An example of a corrected entry is as follows: [https://www.wikidata.org/w/index.php?title=Q105349871&type=revision&diff=1365519277&oldid=1356145998](https://www.wikidata.org/w/index.php?title=Q105349871&type=revision&diff=1365519277&oldid=1356145998)

##### Curation interface 

A web-based (Kotlin, Spring Boot for the back-end, and TypeScript with Vue for the front-end) curation interface is currently in construction.
It will allow mass-editing of entries and navigate quick navigation in the SSOT for the curation of new and existing entries.
This new interface is intended to become open to the public to foster the curation of entries by further means, driven by the users.
In line with the overall LOTUS approach, any modification made in this curation interface will be mirrored after validation on Wikidata and LNPN.


## Code Availability {.page_break_before}

### General Repository

All programs written for this work can be found in the following group: [https://github.com/lotusnprod](https://github.com/lotusnprod).

#### Processing

The source data curation system is available at [https://github.com/lotusnprod/lotus-processor](https://github.com/lotusnprod/lotus-processor).
This program takes the source data as input and outputs curated data, ready for dissemination.
The first step involves checking if the source data has already been processed.
If not, all three elements (biological organism, chemical structures, and references) are submitted to various steps of translation and curation, before validation for dissemination.

#### Wikidata

The programs to interact with Wikidata are available at [https://github.com/lotusnprod/lotus-wikidata-interact](https://github.com/lotusnprod/lotus-wikidata-interact).
On the [upload side](https://github.com/lotusnprod/lotus-wikidata-interact/blob/main/uploadLotus/README.md), the program takes the processed data resulting from the lotusProcessor subprocess as input and uploads it to Wikidata.
It performs a SPARQL query to check which objects already exist.
If needed, it creates the missing objects.
It then updates the content of each object.
Finally, it updates the chemical compound page with a "found in taxon" statement complemented with a "stated in" reference.
A [publication importer](https://github.com/lotusnprod/lotus-wikidata-interact/blob/main/importPublication/README.md) creating an article page from a DOI is also available.

On the [download side](https://github.com/lotusnprod/lotus-wikidata-interact/blob/main/downloadLotus/README.md), the program takes the structured data in Wikidata corresponding to chemical compounds found in taxa with a reference associated as input and exports it in both RDF and tabular formats for further use.
Two subsequent options are (a) that the end-user can directly use the exported data.; or (b) that the exported data, which can be new or modified since the last iteration, is used as new source data in lotusProcessor.

### LNPN 

The LNPN website and processing system is available at [https://github.com/lotusnprod/lotus-web](https://github.com/lotusnprod/lotus-web).
This system takes the processed data resulting from the lotusProcessor as input and uploads it on [https://lotus.naturalproducts.net](https://lotus.naturalproducts.net).
The repository is not part of the main GitHub group as it benefits from already established pipelines developed by CS and MS.
The website allows searches from different points of view, complemented with taxonomies for both the chemical and biological sides.
Many chemical molecular properties and molecular descriptors that otherwise are unavailable in Wikidata are also provided.

### Code Freezing

All repository hyperlinks in the manuscript point to the main branches by default.
The links contain all programs and code and will eventually be updated to a publication branch using modifications resulting from the peer-reviewing process.
As the code evolves, readers are invited to refer to the main branch of each repository for the most up-to-date code.
A frozen version (2021-12-23) of all programs and code is also available in the [LOTUS Zenodo community](https://zenodo.org/communities/the-lotus-initiative) ([5802107](https://doi.org/10.5281/zenodo.5802107) [@doi:10.5281/zenodo.5802107], [5802113](https://doi.org/10.5281/zenodo.5802113) [@doi:10.5281/zenodo.5802113], and [5802120](https://doi.org/10.5281/zenodo.5802120) [@doi:10.5281/zenodo.5802120]).

### Programs and packages {.page_break_before}

#### R
The [R](https://www.r-project.org) versions used for the project were 4.0.2 up to 4.1.2, and R-packages used were, in alphabetical order:
ChemmineR (3.42.1) [@doi:10/b46rs7], chorddiag (0.1.3) [@florChorddiagInteractiveChord2020], ClassyfireR (0.3.6) [@doi:10/gc5tqv], data.table (1.14.2) [@dowleDataTableExtension2020], DBI (1.1.1) [@rspecialinterestgroupondatabasesr-sig-dbDBIDatabaseInterface2021], gdata (2.18.0) [@warnesGdataVariousProgramming2017], ggalluvial (0.12.3) [@doi:10/gg4ghf], ggfittext (0.9.1) [@wilkinsGgfittextFitText2020], ggnewscale (0.4.5) [@campitelliGgnewscaleMultipleFill2021], ggraph (2.0.5) [@pedersenGgraphImplementationGrammar2020], ggstar (1.0.2) [@xuGgstarStarLayer2021], ggtree (3.2.0) [@doi:10/f9qv8x], ggtreeExtra (1.4.0) [@doi:10.1093/molbev/msab166], jsonlite (1.7.2) [@wikidata:Q106204620], pbmcapply (1.5.0) [@kuangPbmcapplyTrackingProgress2019], plotly (4.10.0) [@sievertInteractiveWebBasedData2020], rcrossref(1.1.0.99) [@chamberlainRcrossrefClientVarious2020], readxl (1.3.1) [@wickhamReadxlReadExcel2019], rentrez (1.2.3) [@doi:10/gh5ptg], rotl (3.0.11) [@doi:10/f9jgkm], rvest (1.0.2) [@wickhamRvestEasilyHarvest2020], splitstackshape (1.4.8) [@mahtoSplitstackshapeStackReshape2019], RSQLite (2.2.8) [@mullerRSQLiteSQLiteInterface2021], stringdist (0.9.8) [@doi:10/ghfr9v], stringi (1.7.6) [@gagolewskiPackageStringiCharacter2020], tidyverse (1.3.1) [@doi:10/ggddkj], treeio (1.18.0) [@doi:10/ggwr93], UpSetR (1.4.0) [@gehlenborgUpSetRMoreScalable2019], webchem (1.1.1) [@doi:10/gh5ptf], XML (3.99-0.8) [@langXMLToolsParsing2020], xml2 (1.3.3) [@wickhamXml2ParseXML2020]

#### Python
The [Python](https://www.python.org) version used was 3.7.12 up to 3.9.7, and the Python packages utilized were, in alphabetical order: 
cmcrameri (1.4) [@doi:10.5281/zenodo.1243862; @doi:10.1038/s41467-020-19160-7], faerun (0.3.20) [@doi:10/gcmhc5], map4 (1.0) [@doi:10/gjjz4c], matplotlib (3.5.0) [@doi:10.1109/MCSE.2007.55], [Molvs](https://github.com/mcs07/MolVS) (0.1.1), pandas (1.3.4) [@doi:10/gh6dqz], rdkit (2021.09.2) [@RDKitOpensourceCheminformatics], scipy (1.7.3) [@doi:10/ggj45f], tmap (1.0.4) [@doi:10/gjjz4f].

#### Kotlin
Kotlin packages used were as follows: 
Common: Kotlin 1.4.21 up to 1.6.0, Univocity 2.9.1, OpenJDK 15, Kotlin serialization 1.3.1, konnector 0.1.34, Log4J 2.14.1
Wikidata Importer Bot:, wdkt 0.12.1, CDK 2.5 [@doi:10/gbppnb], RDF4J 3.7.4, Ktor 1.6.5, KotlinXCli 0.3.3, Wikidata data processing: Shadow 5.0.0
Quality control and testing: Ktlint 10.2.0, Kotlinter 3.3.0, Detekt 1.15.0, Ben Mane's version plugin 0.36.0, Junit 5.8.1

#### Additional executable files
[GNFinder](https://github.com/gnames/gnfinder/releases/tag/v0.16.3) v.0.16.3, [GNVerifier](https://github.com/gnames/gnverifier/releases/tag/v0.6.1) v.0.6.1, [OPSIN](https://github.com/dan2097/opsin/releases/tag/2.5.0) v.2.5.0 [@doi:10/b2zkr9]


## Data Availability

A snapshot of the obtained data at the time of re-submission (2021-12-20) is available at the following Zenodo community: [https://zenodo.org/communities/the-lotus-initiative](https://zenodo.org/communities/the-lotus-initiative) (and related records, for data: [5793224](https://zenodo.org/record/5793224) [@doi:10.5281/zenodo.5793224], [5794107](https://zenodo.org/record/5794107) [@doi:10.5281/zenodo.5794107], [5794597](https://zenodo.org/record/5794597) [@doi:10.5281/zenodo.5793224], [5794597](https://zenodo.org/record/5801816) [@doi:10.5281/zenodo.5801816].
The [https://lotus.nprod.net](https://lotus.nprod.net) website is intended to gather news and features related to the LOTUS initiative in the future.


## Acknowledgments {.page_break_before}

AR, JLW and PMA are thankful to the Swiss National Science Foundation for supporting part of this project through the SNF Sinergia grant CRSII5_189921.
JB and AR are really thankful to JetBrains for the Free educational license of IntelliJ and the excellent support received on Youtrack.
JB, JGG, and GFP gratefully acknowledge the support of this work by grant U41 AT008706 and supplemental funding to P50 AT000155 from NCCIH and ODS of the NIH.
MS and CS are supported by the German Research Foundation within the framework ChemBioSys (Project-ID 239748522, SFB 1127).
EW and DM acknowledge the Scholia grant from the Alfred P. Sloan Foundation under grant number G-2019-11458.
The work on the Wikidata IDSM/Sachem endpoint was supported by an ELIXIR CZ research infrastructure project grant (MEYS Grant No: LM2018131) including access to computing and storage facilities.
The authors would like to thank [Dmitry Mozzherin](https://github.com/dimus) for his work done for the Global Names Architecture and related improvements.
They would like to acknowledge help from the PubChem team in integrating LOTUS, especially Dr. Tiejun Cheng and Dr. Evan Bolton.
The authors would also like to thank Layla Michán for starting to add pigment information on Wikidata.
The authors would also like to thank the team behind [Manubot](https://manubot.org/) [@doi:10.1371/journal.pcbi.1007128], used to write this manuscript.
The authors would also like to thank contributors of all electronic NP resources used in this work and the NP community at large.

## Competing interests

The authors declare no competing interest.


## Appendices {.page_break_before}

### Appendix 1 Data Sources List

<div style="font-size:6pt">

Table: Data Sources List {#tbl:si-1}

|database|type|initial retrieved unique entries|cleaned referenced structure-organism pairs|pairs validated for wikidata export|actual validated pairs on wikidata|website|article|retrieval|license status|contact|dump|status|timestamp|
|:------:|:----:|-----:|-----:|-----:|-----:|:----:|:----:|:----:|:------:|:-----:|:---:|:-----:|
|afrotryp|open|313|93|55|54|-|[article](https://link.springer.com/article/10.1007/s00044-016-1764-y?shared-article-renderer) [@doi:10/gh5ptd]|[download](https://static-content.springer.com/esm/art%3A10.1007%2Fs00044-016-1764-y/MediaObjects/44_2016_1764_MOESM1_ESM.doc)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/afrotryp.md)|[Fidele Ntie-Kang](https://link.springer.com/article/10.1007/s00044-016-1764-y/email/correspondent/c1/new) or [Ngozi Justina Nwodo](https://link.springer.com/article/10.1007/s00044-016-1764-y/email/correspondent/c2/new)|YES|unmaintained|2021-09-20T07:35:47Z|
|alkamid|open|4416|2639|2309|2160|[website](http://alkamid.ugent.be)|[article](http://dx.doi.org/10.1016/j.jep.2012.05.038) [@doi:10/gh5ptc]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/alkamid/scraping.R) |[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/alkamid.md)|[Bart De Spiegeleer](mailto:Bart.DeSpiegeleer@UGent.be)|NO|maintained|2021-01-07T07:08:44Z|
|antibase|commercial|46956|45221|-|-|-|-|-|-|-|NO|unmaintained|2021-09-20T07:40:28Z|
|antimarin|commercial|73017|67559|-|-|-|-|-|-|-|NO|unmaintained|2021-09-20T07:41:50Z|
|biofacquim|open|531|519|519|511|[website](https://biofacquim.herokuapp.com) (old version)|[article_old](https://www.mdpi.com/2218-273X/9/1/31) [article_new](https://f1000research.com/articles/8-2071) [@doi:10/gh5ptb]|[download](https://ndownloader.figshare.com/files/20050244)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/biofacquim.md)|[José Medina-Franco](mailto:medinajl@unam.mx)|YES|maintained|2021-09-14T15:28:05Z|
|biophytmol|open|543|558|322|308|[website](http://ab-openlab.csir.res.in/biophytmol)|[article](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-014-0046-2) [@doi:10/f6ss69]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/biophytmol/scraping.R)|[license_CCBY](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/biophytmol.md)|[Anshu Bhardwaj](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-014-0046-2/email/correspondent/c1/new)|NO|unmaintained|2021-09-14T15:36:13Z|
|carotenoiddb|open|2922|639|530|485|[website](http://carotenoiddb.jp/About.html)|[article](https://doi.org/10.1093/database/bax004) [@doi:10/f9x5p2] |[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/carotenoiddb/scraping.R)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/carotenoiddb.md)|[yzjunko@gmail.com](mailto:yzjunko@gmail.com)|NO|maintained|2021-01-07T07:08:45Z|
|coconut|open|5757872|5723691|153981|140877|[website](https://coconut.naturalproducts.net)|[article](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00424-9) [@doi:10/ggwwcq]|[download](https://zenodo.org/record/3778405/files/COCONUT.sdf.zip?download=1)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/coconut.md)|[Maria Sorokina](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00424-9/email/correspondent/c1/new)|YES|maintained|2021-09-23T21:27:39Z|
|cyanometdb|open|1905|1631|1621|1605|-|[article](https://doi.org/10.1016/j.watres.2021.117017)  [@doi:10/gjgdzm]|[download](https://ars.els-cdn.com/content/image/1-s2.0-S0043135421002153-mmc4.zip)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/cyanometdb.md)|[elisabeth.janssen@eawag.ch](mailto:elisabeth.janssen@eawag.ch)|YES|maintained|2021-09-14T13:16:33Z|
|datawarrior|open|589|541|71|60|[website](http://www.openmolecules.org/datawarrior/index.html)|[article](https://doi.org/10.1021/ci500588j) [@doi:10/f637rz]|[download](http://www.openmolecules.org/datawarrior/download.html)|[no_license](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/datawarrior.md)|[thomas.sander@idorsia.com](mailto:thomas.sander@idorsia.com)|YES|retired|2021-01-07T07:08:45Z|
|dianatdb|open|290|323|115|111|[website](http://rdu.iquimica.unam.mx/handle/20.500.12214/1186)|[article](https://doi.org/10.1039/D0RA10453A) [@doi:10.1039/D0RA10453A]|[download](http://rdu.iquimica.unam.mx/bitstream/20.500.12214/1186/3/2020_DiaNatDB_336.xlsx)|[license_CCBY_NC](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/dianatdb.md)|[amadariaga@iquimica.unam.mx](mailto:amadariaga@iquimica.unam.mx) or [kmtzm@unam.mx](mailto:kmtzm@unam.mx)|YES|maintained|2021-08-16T21:43:20Z|
|dnp|commercial|205072|254573|-|-|[website](http://dnp.chemnetbase.com/faces/chemical/ChemicalSearch.xhtml)|-|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/dnp/extracting.py)|-|[support@taylorfrancis.com](mailto:support@taylorfrancis.com)|NO|maintained|2021-11-02T10:55:14Z|
|drduke|open|90675|9660|6184|5222|[website](https://phytochem.nal.usda.gov/phytochem/help/index/about)|-|[download](https://data.nal.usda.gov/system/files/Duke-Source-CSV.zip)|[license_CC0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/drduke.md)|[agref@usda.gov](mailto:agref@usda.gov)|YES|maintained|2021-11-02T10:55:16Z|
|foodb|restricted|81941|39662|-|-|[website](https://foodb.ca/about#cite)|-|[download](https://foodb.ca/public/system/downloads/foodb_2020_4_7_csv.tar.gz)|[license_CCBY_NC](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/foodb.md)|[jreid3@ualberta.ca](mailto:jreid3@ualberta.ca) (Jennifer)|YES|unmaintained|2021-11-02T10:55:17Z|
|inflamnat|open|665|632|306|268|-|[article](https://pubs.acs.org/doi/10.1021/acs.jcim.8b00560) [@doi:10/gfs7x6]|[download](https://pubs.acs.org/doi/suppl/10.1021/acs.jcim.8b00560/suppl_file/ci8b00560_si_001.xlsx)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/inflamnat.md)|[xiaoweilie@ynu.edu.cn](mailto:xiaoweilie@ynu.edu.cn)|YES|unmaintained|2021-09-14T13:51:30Z|
|knapsack|open|132127|139336|59945|55186|[website](http://www.knapsackfamily.com/knapsack_core/top.php)|[article](https://link.springer.com/chapter/10.1007/3-540-29782-0_13) [@doi:10.1007/3-540-29782-0_13; @doi:10/cstrnq]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/knapsack/scraping.R)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/knapsack.md)|[skanaya@gtc.naist.jp](mailto:skanaya@gtc.naist.jp)|NO|maintained|2021-09-17T10:15:34Z|
|metabolights|open|38208|37704|6241|5687|[website](https://www.ebi.ac.uk/metabolights/about)|[article](https://doi.org/10.1093/nar/gkz1019) [@doi:10.1093/nar/gkz1019]|[download](ftp://ftp.ebi.ac.uk/pub/databases/metabolights/eb-eye/eb-eye_metabolights_complete.xml)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/metabolights.md)|-|YES|maintained|2021-09-14T13:53:55Z|
|mibig|open|1310|1139|638|535|[website](https://mibig.secondarymetabolites.org)|[article](https://doi.org/10.1093/nar/gkz882) [@doi:10.1093/nar/gkz882]|[download](https://dl.secondarymetabolites.org/mibig/mibig_json_2.0.tar.gz)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/mibig.md)|[Tilmann Weber](mailto:tiwe@biosustain.dtu.dk) or[Marnix Medema](mailto:marnix.medema@wur.nl)|YES|unmaintained|2021-09-23T20:59:48Z|
|mitishamba|open|1071|534|294|291|[website](http://mitishamba.uonbi.ac.ke)|[article](http://erepository.uonbi.ac.ke/handle/11295/92273) [@deresesolomonMitishambaDatabaseWeb2015]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/mitishamba/scraping.R)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/mitishamba.md)|-|NO|defunct|2020-06-12T13:34:39Z|
|nanpdb|open|5752|6383|5937|5283|[website](http://african-compounds.org/nanpdb)|[article](https://doi.org/10.1021/acs.jnatprod.7b00283) [@doi:10/ggfdw7]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/nanpdb/scraping.R)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/nanpdb.md)|[ntiekfidele@gmail.com](mailto:ntiekfidele@gmail.com) [stefan.guenther@pharmazie.uni-freiburg.de](mailto:stefan.guenther@pharmazie.uni-freiburg.de)|NO|maintained|2020-06-12T13:43:10Z|
|napralert|commercial|681401|392498|294818|270743|[website](https://www.napralert.org)|[article](https://doi.org/10.1016/B978-008045382-8.00060-5) [@doi:10.1016/b978-008045382-8.00060-5]|-|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/napralert.md)|[napralert@uic.edu](mailto:napralert@uic.edu)|NO|defunct|2020-12-31T21:47:44Z|
|npass|open|290535|30185|25429|23612|[website](http://bidd2.nus.edu.sg/NPASS/about.php)|[article](https://academic.oup.com/nar/article/46/D1/D1217/4584628) [@doi:10/gh5ptk]|[download](http://bidd.group/NPASS/downloadnpass.html)|[license_CCBY_NC](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/npass.md)|[phacyz@nus.edu.sg](mailto:phacyz@nus.edu.sg) [jiangyy@sz.tsinghua.edu.cn](mailto:jiangyy@sz.tsinghua.edu.cn) [iaochen@163.com](mailto:iaochen@163.com)|YES|unmaintained|2020-07-10T13:03:09Z|
|npatlas|open|32539|34726|34548|33087|[website](https://www.npatlas.org/joomla)|[article](https://doi.org/10.1021/acscentsci.9b00806) [@doi:10/gh5ptm; @doi:10.1093/nar/gkab941]|[download](https://www.npatlas.org/custom/versions/np_atlas_2020_06/download.tsv)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/npatlas.md)|[rliningt@sfu.ca](mailto:rliningt@sfu.ca)|YES|maintained|2021-08-18T18:29:55Z|
|npcare|open|7763|5878|3790|3538|[website](http://silver.sejong.ac.kr/npcare)|[article](https://doi.org/10.1186/s13321-016-0188-5) [@doi:10/f9rhrj]|[download](http://silver.sejong.ac.kr/npcare/csv/npcare.zip)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/npcare.md)|[choihwanho@gmail.com](mailto:choihwanho@gmail.com)|YES|unmaintained|2016-12-20T12:40:12Z|
|npedia|open|82|99|28|28|[website](http://www.cbrg.riken.jp/npedia/?LANG=en)|[article](https://www.jstage.jst.go.jp/article/ciqs/2006/0/2006_0_JL6/_article) [@inenglishplease]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/npedia/scraping.R)|[no_license](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/npedia.md)|[hisyo@riken.jp](mailto:hisyo@riken.jp) [npd@riken.jp](mailto:npd@riken.jp)|NO|defunct|2020-06-03T08:23:04Z|
|nubbe|open|2189|2340|2340|2119|[website](https://nubbe.iq.unesp.br/portal/nubbe-search.html)|[article](https://www.nature.com/articles/s41598-017-07451-x) [@doi:10/gh5ps7]|-|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/nubbe.md)|[Vanderlan S. Bolzani](https://www.nature.com/articles/s41598-017-07451-x/email/correspondent/c1/new)|NO|maintained|2021-11-02T10:55:18Z|
|pamdb|open|3046|2820|24|24|[website](http://pseudomonas.umaryland.edu/PAMDB.htm)|[article](https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkx1061) [@doi:10/gh5ps9]|[download](http://pseudomonas.umaryland.edu/PaDl/PaMet.xlsx)|[license_CCBY_NC](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/pamdb.md)|[awilks@rx.umaryland.edu](mailto:awilks@rx.umaryland.edu) [aoglesby@rx.umaryland.edu](mailto:aoglesby@rx.umaryland.edu) [mkane@rx.umaryland.edu](mailto:mkane@rx.umaryland.edu)|YES|unmaintained|2017-03-02T19:47:50Z|
|phenolexplorer|open|8077|8700|7123|5721|[website](http://phenol-explorer.eu/cite_us)|[article](http://database.oxfordjournals.org/content/2013/bat070.long) [@doi:10/gh5ps6]|[download](http://phenol-explorer.eu/downloads)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/phenolexplorer.md)|[scalberta@iarc.fr](mailto:scalberta@iarc.fr)|YES|unmaintained|2016-12-11T02:27:57Z|
|phytohub|open|2349|1145|132|94|[website](http://phytohub.eu/about)|[article](https://hal.archives-ouvertes.fr/hal-01697081/file/2017_Giacomoni_ICPH_Québec.pdf) [@https://hal.archives-ouvertes.fr/hal-01697081]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/phytohub/scraping.R)|[no_license](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/phytohub.md)|[claudine.manach@inra.fr](mailto:claudine.manach@inra.fr)|YES|unmaintained|2020-06-03T08:45:20Z|
|procardb|open|6556|6278|60|55|[website](https://bioinfo.imtech.res.in/servers/procardb)|[article](https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6) [@doi:10/f8wjvh]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/procardb/scraping.R)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/procardb.md)|[Anil Kumar Pinnaka](https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6/email/correspondent/c1/new) [Ashwani Kumar](https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6/email/correspondent/c2/new)|NO|unmaintained|2020-08-10T19:00:42Z|
|respect|open|2759|1064|634|547|[website](http://spectra.psc.riken.jp/menta.cgi/respect/licence/licence)|[article](https://www.sciencedirect.com/science/article/pii/S003194221200310X) [@doi:10/f36bkc]|[download](http://spectra.psc.riken.jp/menta.cgi/static/respect/respect.zip)|[license_CCBY_NC_2.1_Japan](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/respect.md)|[ksaito@psc.riken.jp](mailto:ksaito@psc.riken.jp)|YES|unmaintained|2021-11-02T10:55:18Z|
|sancdb|open|860|925|747|732|[website](https://sancdb.rubi.ru.ac.za/cite_us)|[article](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0080-8) [@doi:10/f7gjs6]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/sancdb/scraping.R)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/sancdb.md)|[Özlem Tastan Bishop](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0080-8/email/correspondent/c1/new)|NO|unmaintained|2020-06-03T09:48:30Z|
|streptomedb|open|71638|33217|20715|18395|[website](http://132.230.56.4/streptomedb3)|[article](https://doi.org/10.1093/nar/gkv1319) [@doi:10/f8cnnk]|[download](http://132.230.56.4/streptomedb/download)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/streptomedb.md)|[stefan.guenther@pharmazie.uni-freiburg.de](mailto:stefan.guenther@pharmazie.uni-freiburg.de)|YES|maintained|2021-10-03T22:39:24Z|
|swmd|open|1075|1751|1597|1479|[website](http://www.swmd.co.in/index.php)|[article](https://dx.doi.org/10.6026%2F97320630005361) [@doi:10/gh5ps5]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/swmd/scraping.R)|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/swmd.md)|[Dicky.John@gmail.com](mailto:Dicky.John@gmail.com)|NO|unmaintained|2021-11-02T10:55:18Z|
|tmdb|open|2116|533|26|24|[website](http://pcsb.ahau.edu.cn:8080/TCDB/index.jsp)|[article](https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1) [@doi:10/f6hnfz]|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/tmdb/scraping.R)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/tmdb.md)|[Xiao-Chun Wan](https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1/email/correspondent/c1/new) [Guan-Hu Bao](https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1/email/correspondent/c2/new)|NO|unmaintained|2020-06-03T10:04:22Z|
|tmmc|open|15033|7833|5826|4015|[website](http://informatics.kiom.re.kr/compound/index.jsp)|[article](https://bmccomplementmedtherapies.biomedcentral.com/articles/10.1186/s12906-015-0758-5) [@doi:10.1186/s12906-015-0758-5]|[download](http://informatics.kiom.re.kr/compound/download/compound.xlsx)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/tmmc.md)|[Jeong-Ju Lee](https://bmccomplementmedtherapies.biomedcentral.com/articles/10.1186/s12906-015-0758-5/email/correspondent/c1/new)|YES|unmaintained|2016-06-16T04:28:00Z|
|tppt|open|27182|23872|684|641|[website](https://www.agroscope.admin.ch/agroscope/en/home/publications/apps/tppt.html)|[article](https://pubs.acs.org/doi/10.1021/acs.jafc.8b01639) [@doi:10/gd3bm6]|[download](https://www.agroscope.admin.ch/dam/agroscope/de/dokumente/publikationen/tppt-xls.xlsx.download.xlsx/TPPT_database.xlsx)|[license_copyright](https://github.com/lotusnprod/lotus-processor/blob/main/doc/licenses/tppt.md)|[thomas.bucheli@agroscope.admin.ch](mailto:thomas.bucheli@agroscope.admin.ch)|YES|unmaintained|2021-09-24T19:50:21Z|
|unpd|open|331242|304683|211158|197710|[website](https://web.archive.org/web/20160314180639/http://pkuxxj.pku.edu.cn/UNPD/introduction.php)|[article](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0062839) [@doi:10/f43xwq]|-|[license_CCBY_4.0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/unpd.md)|[lirongc@pku.edu.cn](mailto:lirongc@pku.edu.cn) [xiaojxu@pku.edu.cn](mailto:xiaojxu@pku.edu.cn)|NO|defunct|2021-09-14T16:41:31Z|
|wakankensaku|open|367|224|208|202|[website](https://wakankensaku.inm.u-toyama.ac.jp/wiki/Main_Page)|-|[script](https://github.com/lotusnprod/lotus-processor/blob/main//src/1_gathering/db/wakankensaku/scraping.R)|-|-|NO|defunct|2020-07-02T08:35:05Z|
|wikidata|open|951268|960611|959747|919752|[website](https://www.wikidata.org/wiki/Wikidata:Main_Page)|-|[download](https://w.wiki/3JpE)|[license_CC0](https://github.com/lotusnprod/lotus-processor/blob/main/docs/licenses/wikidata.md)|-|YES|maintained|-|

</div>


### Appendix 2 Summary of the Validation Statistics {.page_break_before}

<div style="font-size:10pt">

Table: Summary of the Validation Statistics {#tbl:si-2}

| | **First validation dataset (n =420)** <!-- $colspan="8" --> | | | | | | | | **Second validation dataset (n = 100)** <!-- $colspan="2" --> | |
|:----:|----:|----:|----:|----:|----:|----:|---:|---:|----:|----:|
| **Reference Type ** | **True positive** | **False positive** |**False negative** |**True negative** |**Ratio** |**Precision** | **Recall** |**F~0.5~ score** |**True positive** |**False negative** |
| **Original** | 80 | 6 | 7 | 11 | 0.31 | 0.93 | 0.92 | 0.92 | 38 | 1 |
| **Pubmed** | 37 | 1 | 5 | 6 | 0.30 | 0.97 | 0.88 | 0.92 | 5 | 1 |
| **DOI** | 115 | 6 | 0 | 6 | 0.19 | 0.95 | 1.00 | 0.97 | 43 | 1 |
| **Title** | 38 | 2 | 0 | 16 | 0.12 | 0.95 | 1.00 | 0.97 | 7 | 0 |
| **Split** | 8 | 0 | 15 | 27 | 0.08 | 1.00 | 0.35 | 0.52 | 4 | 0 |
| **Publishing details** | 1 | 0 | 1 | 32 | 0.01 | 1.00 | 0.50 | 0.67 | 0 | 0 |
| **Total** | 279 | 15 | 28 | 98 | 1.00 | - | - | - | **97** | **3** |
| **Corrected total** | - | - | - | - | - | **0.96** | **0.89** | **0.91** | - | - |

</div>


### Appendix 3 Wikidata Queries {.page_break_before}

#### Query 1 - *Arabidopsis thaliana*

This query answers to the following question:

```BASH
What are the compounds found in Mouse-ear cress Arabidopsis thaliana (Q158695) or children taxa?
```

Link: [https://w.wiki/4Vcv](https://w.wiki/4Vcv)
```SQL
#title: What are the compounds found in Mouse-ear cress Arabidopsis thaliana (Q158695) or children taxa?
SELECT DISTINCT ?structure ?structureLabel ?structure_inchi WHERE {
  VALUES ?taxon {
    wd:Q158695                           # You can remove the Qxxxxxx and hit Ctrl+space, type the first letters and it should autocomplete
  }
  ?children (wdt:P171*) ?taxon.          # Include children taxa
  ?structure wdt:P234 ?structure_inchi;  # Get the InChI
    (p:P703/ps:P703) ?children.          # Found in given taxon/taxa
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
```

#### Query 2 - *β*-sitosterol {.page_break_before}

This query answers to the following question:

```BASH
Which organisms are known to contain β-sitosterol?
```

Link: [https://w.wiki/4VFn](https://w.wiki/4VFn)
```SQL
#title: Which organisms are known to contain Beta-Sitosterol (Q121802)?
SELECT DISTINCT ?taxon ?taxon_name WHERE {
  VALUES ?compound {
    wd:Q121802                       # You can remove the Qxxxxxx and hit Ctrl+space, type the first letters and it should autocomplete
  }
  ?compound (p:P703/ps:P703) ?taxon. # Found in taxon
  ?taxon wdt:P225 ?taxon_name.       # Get scientific name of the taxon
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
```

#### Query 3 - *β*-sitosterol stereoisomers {.page_break_before}

This query answers to the following question:

```BASH
Which organisms are known to contain which stereoisomers of β-sitosterol?
```

Link: [https://w.wiki/4VFq](https://w.wiki/4VFq)
```SQL
#title: Which organisms are known to contain stereoisomers of Beta-Sitosterol (Q121802)?
SELECT ?taxon_name ?compound ?InChIKey 
WITH {
  SELECT ?compound ?InChIKey WHERE {
    VALUES ?target {
      wd:Q121802
    }
    ?target wdt:P235 ?queryKey.
    ?compound wdt:P235 ?InChIKey.
    FILTER (REGEX(STR(?InChIKey), CONCAT("^", SUBSTR($queryKey,1,14), "-")))
    FILTER (?InChIKey != ?queryKey)
  }
} AS %compounds
WHERE {
  INCLUDE %compounds
          ?compound (wdt:P703/wdt:P225) ?taxon_name.
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
```

#### Query 4 - Pigments {.page_break_before}

This query answers to the following question:

```BASH
Which pigments are found in which taxa, according to which reference?
```

Link: [https://w.wiki/4VFx](https://w.wiki/4VFx)
```SQL
#title: Which pigments are found in which taxa, according to which reference?
# special thanks goes to User:Lmichan for updating this information!
SELECT DISTINCT ?compound ?compoundLabel ?taxon ?taxonname ?DOI 
WITH {
  SELECT ?compound WHERE {
    ?compound (wdt:P31*/wdt:P279*) wd:Q161179.  # get pigments
  }
} AS %compounds
WITH {
  SELECT ?compound ?P703statement WHERE {
    INCLUDE %compounds
            ?compound p:P703 ?P703statement.    # check for "found in taxon" statements
  }
} AS %P703statement
WITH {
  SELECT ?compound ?taxon ?DOI WHERE {
    INCLUDE %P703statement
            ?P703statement ps:P703 ?taxon ;     # get the respective taxa
            prov:wasDerivedFrom / pr:P248 [     # get the reference supporting that statement
              wdt:P356 ?DOI                     # get the DOI for the reference
            ] .
  }
} AS %taxa
WHERE {
  {
    INCLUDE %taxa

            ?taxon wdt:P225 ?taxonname .        # get the taxon name
  }
  ?compound rdfs:label ?compoundLabel .         # get compound labels
  FILTER (LANG(?compoundLabel) = "en") .        # filter for English
}
ORDER BY ASC(?compoundLabel)
LIMIT 10000
```

#### Query 5 - Sister taxon compounds {.page_break_before}

This query answers to the following question:

```BASH
What are examples of organisms where compounds were found in an organism sharing the same parent taxon, but not in the organism itself?
```


Link: [https://w.wiki/4Wt3](https://w.wiki/4Wt3)
```SQL
#title: What are examples of organisms where compounds were found in an organism sharing the same parent taxon, but not the organism itself?
SELECT DISTINCT ?compound ?compoundLabel ?taxonname_with_compound ?taxonname_without_compound ?parent_taxon WITH{ 
  SELECT DISTINCT ?compound ?taxon_with_compound ?parent_taxon 
  WHERE {
    ?compound wdt:P235 ?inchikey.
    SERVICE bd:sample { ?compound wdt:P703 ?taxon_with_compound. bd:serviceParam bd:sample.limit 1000 }   
    ?taxon_with_compound wdt:P171 ?parent_taxon.
  }
} AS %taxon_with_compound 
WITH
{ 
  SELECT DISTINCT ?taxon_without_compound ?parent_taxon ?compound 
  WHERE {
    INCLUDE %taxon_with_compound
    ?taxon_without_compound wdt:P171 ?parent_taxon.
    FILTER (?taxon_with_compound != ?taxon_without_compound)
  }
} AS %taxon2 
WHERE {
  INCLUDE %taxon_with_compound
  INCLUDE %taxon2
  FILTER NOT EXISTS {?compound wdt:P703 ?taxon_without_compound.}
  ?taxon_with_compound wdt:P225 ?taxonname_with_compound.
  ?taxon_without_compound wdt:P225 ?taxonname_without_compound.
  ?compound rdfs:label ?compoundLabel.
  FILTER(LANG(?compoundLabel) = "en").
}
```

#### Query 6 - *Zephyranthes* sister taxon compounds {.page_break_before}

This query answers to the following question:

```BASH
Which Zephyranthes spp. lack compounds known from ≥ 2 other Zephyranthes?
```

Link: [https://w.wiki/4VG3](https://w.wiki/4VG3)
```SQL
#title: Which Zephyranthes (Q191364) spp. lack compounds known from at least two species in the genus?
PREFIX target: <http://www.wikidata.org/entity/Q191364> # Zephyranthes
SELECT DISTINCT ?compound ?compoundLabel ?taxon_with_compound ?another_taxon_with_compound ?taxon_without_compound WITH { 
  SELECT DISTINCT ?compound ?taxon_YES_1 ?taxon_YES_2 
  WHERE {
    ?compound wdt:P703 ?taxon_YES_1 .
    ?compound wdt:P703 ?taxon_YES_2 .
    ?taxon_YES_1 wdt:P171 target: .
    ?taxon_YES_2 wdt:P171 target: .
    FILTER (?taxon_YES_2 != ?taxon_YES_1)
  }
} AS %taxa_with_compound 
WITH
{ 
  SELECT DISTINCT ?taxon_NO ?compound 
  WHERE {
    INCLUDE %taxa_with_compound
    ?taxon_NO wdt:P171 target: .
    FILTER (?taxon_YES_1 != ?taxon_NO)
  }
} AS %taxon_without_compond 
WHERE {
  INCLUDE %taxa_with_compound
  INCLUDE %taxon_without_compond
  FILTER NOT EXISTS { ?compound wdt:P703 ?taxon_NO .}
  VALUES ?classes {
    wd:Q11173
    wd:Q59199015
  }
  ?taxon_YES_1 wdt:P225 ?taxon_with_compound .
  ?taxon_YES_2 wdt:P225 ?another_taxon_with_compound .
  ?taxon_NO wdt:P225 ?taxon_without_compound .
  ?compound (wdt:P31*/wdt:P279*) ?classes .
  ?compound rdfs:label ?compoundLabel.
  FILTER(LANG(?compoundLabel) = "en").
}
```

#### Query 7 - Antibiotic-like compounds {.page_break_before}

This query answers to the following question:

```BASH
How many compounds are structurally similar to compounds labeled as antibiotics? Results are grouped by the parent taxon of the organism they were found in.
```

Link: [https://w.wiki/4VG4](https://w.wiki/4VG4)
```SQL
#title: How many compounds are structurally similar to compounds labeled as antibiotics? Results are grouped by the parent taxon of the organism they were found in.
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#> # prefixes needed for structural similarity search
PREFIX idsm: <https://idsm.elixir-czech.cz/sparql/endpoint/>
SELECT ?parent_taxon ?parent_taxon_name (COUNT(DISTINCT ?compound) AS ?count) WHERE {
  SERVICE idsm:wikidata {
    VALUES ?CUTOFF {
      "0.9"^^xsd:double
    }
    SERVICE <https://query.wikidata.org/bigdata/namespace/wdq/sparql> {
      VALUES ?MESH {
        "D000900"
      }
      ?antibiotic ((wdt:P279*)/wdt:P2868/wdt:P486) ?MESH;
        wdt:P233 ?smiles.
    }
    ?compound sachem:similarCompoundSearch _:b40.
    _:b40 sachem:query ?smiles;
      sachem:cutoff ?CUTOFF.
  }
  hint:Prior hint:runFirst "true"^^xsd:boolean.
  ?compound wdt:P703 ?taxon.
  ?taxon wdt:P171 ?parent_taxon.
  OPTIONAL { ?parent_taxon wdt:P225 ?parent_taxon_name. }
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
GROUP BY ?parent_taxon ?parent_taxon_name
ORDER BY DESC (?count)
```

#### Query 8 - Indolic scaffold {.page_break_before}

This query answers to the following question:

```BASH
Which organisms contain indolic scaffolds? Count occurrences, group and order the results by the parent taxon.
```

Link: [https://w.wiki/4VG9](https://w.wiki/4VG9)
```SQL
#title: Which organisms contain indolic scaffolds? Count occurences, group and order the results by the parent taxon.
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#> # prefixes needed for structural similarity search
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX p: <http://www.wikidata.org/prop/>
PREFIX idsm: <https://idsm.elixir-czech.cz/sparql/endpoint/>
SELECT ?parent_taxon ?parent_taxon_name (COUNT(DISTINCT ?compound) AS ?count) WHERE {
  SERVICE idsm:wikidata {
    VALUES ?SUBSTRUCTURE {
      "NCCC1=CNC2=C1C=CC=C2" # indolic scaffold
    }
    ?compound sachem:substructureSearch _:b16.
    _:b16 sachem:query ?SUBSTRUCTURE.
  }
  hint:Prior hint:runFirst "true"^^xsd:boolean.
  ?compound p:P703 ?statement;
    wdt:P235 ?inchikey.
  ?statement ps:P703 ?taxon.
  ?taxon wdt:P171 ?parent_taxon.
  ?parent_taxon wdt:P225 ?parent_taxon_name.
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
GROUP BY ?parent_taxon ?parent_taxon_name
ORDER BY DESC (?count)
```

#### Query 9 - Bioactive compounds in Actinobacteria {.page_break_before}

This query answers to the following question:

```BASH
Which compounds with known bioactivities were isolated from Actinobacteria, between 2014 and 2019, with related organisms and references?
```

Link: [https://w.wiki/4VGC](https://w.wiki/4VGC)
```SQL
#title: Which compounds with known bioactivities were isolated from Actinobacteria (Q26262282), between 2014 and 2019, with related organisms and references?
SELECT ?organism ?organism_name ?compound ?compound_smiles (GROUP_CONCAT(DISTINCT ?meshLabel; SEPARATOR = "|") AS ?bioactivities) ?isolation_reference ?reference_title WHERE {
  ?organism (wdt:P171*) wd:Q26262282;
    wdt:P225 ?organism_name.
  ?compound wdt:P235 ?compound_id;
    wdt:P233 ?compound_smiles;
    p:P703 ?statement;
    (wdt:P2868/wdt:P486) ?meshId.
  ?mesh wdt:P486 ?meshId;
    rdfs:label ?meshLabel.
  FILTER(LANGMATCHES(LANG(?meshLabel), "EN"))
  ?statement ps:P703 ?organism;
    prov:wasDerivedFrom ?ref.
  ?ref pr:P248 ?isolation_reference.
  ?isolation_reference wdt:P1476 ?reference_title;
    wdt:P356 ?reference_doi;
    wdt:P577 ?reference_date.
  FILTER(((YEAR(?reference_date)) >= 2014 ) && ((YEAR(?reference_date)) <= 2019 ))
}
GROUP BY ?organism ?organism_name ?compound ?compound_smiles ?isolation_reference ?reference_title
LIMIT 100000
```

#### Query 10 - *Aspergillus* spp. terpenoids {.page_break_before}

This query answers to the following question:

```BASH
Which compounds labeled as terpenoids were found in Aspergillus spp., between 2010 and 2020, with related references?
```

Link: [https://w.wiki/4VGD](https://w.wiki/4VGD)
```SQL
#title: Which compounds labelled as terpenoid (Q426694) were found in Aspergillus (Q335130) spp., between 2010 and 2020, with related references?
SELECT ?compound ?compound_inchi (GROUP_CONCAT(DISTINCT ?isolation_reference; SEPARATOR = "|") AS ?isolation_references) (GROUP_CONCAT(DISTINCT ?reference_title; SEPARATOR = "|") AS ?references_titles) WHERE {
  VALUES ?taxon {
    wd:Q335130
  }
  VALUES ?chemical_class {
    wd:Q426694
  }
  ?organism (wdt:P171*) ?taxon.
  ?compound wdt:P235 ?compound_id;
    wdt:P234 ?compound_inchi;
    ((wdt:P31|wdt:P279)/(wdt:P279*)) ?compound_class;
    p:P703 ?statement.
  ?statement ps:P703 ?organism;
    (prov:wasDerivedFrom/pr:P248) ?isolation_reference.
  ?isolation_reference wdt:P1476 ?reference_title;
    wdt:P356 ?reference_doi;
    wdt:P577 ?reference_date.
  FILTER(((YEAR(?reference_date)) >= 2010 ) && ((YEAR(?reference_date)) <= 2020 ))
  FILTER(?compound_class = ?chemical_class)
}
GROUP BY ?compound ?compound_inchi
```

#### Query 11 - Triples {.page_break_before}

This query answers to the following question:

```BASH
Which are the available referenced structure-organism pairs on Wikidata? (example limited to 1000 results)
```

Link: [https://w.wiki/4VFh](https://w.wiki/4VFh)
```SQL
#title: Which are the available referenced structure-organism pairs on Wikidata? (example limited to 1000 results)
SELECT DISTINCT ?structure ?structure_inchikey ?taxon ?taxon_name ?reference ?reference_doi WHERE {
  ?structure wdt:P235 ?structure_inchikey;       # get the inchikey
    p:P703[                                      # statement found in taxon
     ps:P703 ?taxon;                             # get the taxon
     (prov:wasDerivedFrom/pr:P248) ?reference ]. # get the reference
  ?taxon wdt:P225 ?taxon_name.                   # get the taxon scientific name
  ?reference wdt:P356 ?reference_doi.            # get the reference DOI
}
LIMIT 1000
```


### Appendix 4 Wikidata Entry Creation Tutorial {.page_break_before}

[
<i class="fas fa-scroll fa-lg"></i> **Tutorial for manual creation**<br>
also check [https://lotus.nprod.net/post/tuto-wd-entry-creation](https://lotus.nprod.net/post/tuto-wd-entry-creation) for the latest version of this tutorial.
]{.banner .lightblue}

#### Manual addition of a referenced structure-organism pair to Wikidata

##### Select a referenced structure-organism pair

Throughout this demonstration, we are going to use the following example:
> [trigocherrin A](https://pubchem.ncbi.nlm.nih.gov/compound/101556657) is found in [_Trigonostemon cherrieri_](http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&commit=Search), as stated in [Trigocherrin A, the first natural chlorinated daphnane diterpene orthoester from Trigonostemon cherrieri](https://doi.org/10.1021/OL2030907).

##### Fetch the information for the referenced structure-organism pair

###### Structure

Search PubChem for your compound, here [trigocherrin A](https://pubchem.ncbi.nlm.nih.gov/#query=%22trigocherrin%20A%22).
This leads to [https://pubchem.ncbi.nlm.nih.gov/compound/101556657](https://pubchem.ncbi.nlm.nih.gov/compound/101556657).

![](images/tutorial/2021-02-25-14-15-54.png){width="100%"}

From there, you can fetch the compound's name, InChIKey and InChI as well as its Canonical and Isomeric SMILES.
Here we keep, respectively:

```BASH
- trigocherrin A
- QOVGHDRCAGYGEB-FFZYJECLSA-N
- InChI=1S/C38H36Cl2O12/c1-18(2)35(44)27-19(3)37-25-16-24(31(39)40)28(48-32(43)22-12-8-6-9-13-22)36(25,45)33(47-21(5)42)34(17-46-20(4)41)29(49-34)26(37)30(35)51-38(50-27,52-37)23-14-10-7-11-15-23/h6-16,19,26-30,33,44-45H,1,17H2,2-5H3/t19-,26+,27?,28+,29+,30-,33-,34+,35+,36-,37+,38?/m1/s1
- CC1C2C(C3C4C1(C5=CC(=C(Cl)Cl)C(C5(C(C6(C4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O2)(O3)C8=CC=CC=C8)(C(=C)C)O
- C[C@@H]1C2[C@]([C@H]3[C@H]4[C@]1(C5=CC(=C(Cl)Cl)[C@@H]([C@]5([C@@H]([C@@]6([C@H]4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O3)(O2)C8=CC=CC=C8)(C(=C)C)O
```

###### Organism

You can check if your organism name is correctly spelled using the Global Names resolver service: [http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&commit=Search](http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&commit=Search).

![](images/tutorial/2021-02-27-18-40-20.png){width="100%"}

Alternatively, you can use [gnfinder](https://github.com/gnames/gnfinder) in your command line interface to check for the spelling of your organism string.
 
```BASH
echo "Trigonostemion cherrieri" | gnfinder find -c -l eng
{
  "metadata": {
    "date": "2021-02-27T18:44:41.640982+01:00",
    "gnfinderVersion": "v0.11.1",
    "withBayes": true,
    "tokensAround": 0,
    "language": "eng",
    "detectLanguage": false,
    "totalWords": 2,
    "totalCandidates": 1,
    "totalNames": 1
  },
  "names": [
    {
      "cardinality": 2,
      "verbatim": "Trigonostemion cherrieri",
      "name": "Trigonostemion cherrieri",
      "odds": 77581.46698350731,
      "start": 0,
      "end": 24,
      "annotationNomenType": "NO_ANNOT",
      "annotation": "",
      "verification": {
        "bestResult": {
          "dataSourceId": 1,
          "dataSourceTitle": "Catalogue of Life",
          "taxonId": "1575885",
          "matchedName": "Trigonostemon cherrieri Veillon",
          "matchedCardinality": 2,
          "matchedCanonicalSimple": "Trigonostemon cherrieri",
          "matchedCanonicalFull": "Trigonostemon cherrieri",
          "classificationPath": "Plantae|Tracheophyta|Magnoliopsida|Malpighiales|Euphorbiaceae|Trigonostemon|Trigonostemon cherrieri",
          "classificationRank": "kingdom|phylum|class|order|family|genus|species",
          "classificationIds": "3939764|3942634|3942724|3942777|3942795|4210752|1575885",
          "editDistance": 1,
          "stemEditDistance": 1,
          "matchType": "FuzzyCanonicalMatch"
        },
        "dataSourcesNum": 13,
        "dataSourceQuality": "HasCuratedSources",
        "retries": 1
      }
    }
  ]
}
```

For misspellings like _Trigonstemion cherrieri_, gnfinder can help resolve them, in this case to _Trigonostemon cherrieri_.


###### Reference

Make sure that you have the correct [Digital Object Identifier (DOI)](https://www.doi.org/doi_handbook/Glossary.html#doi) for it.
For "[Trigocherrin A, the first natural chlorinated daphnane diterpene orthoester from Trigonostemon cherrieri](https://doi.org/10.1021/OL2030907)", this is  **10.1021/ol2030907**.
Note that DOIs are uppercase-normalized in Wikidata.

##### Check for the presence of your compound in Wikidata <a name="check_compound"></a>

Using the compound's InChIKey (i.e. ```QOVGHDRCAGYGEB-FFZYJECLSA-N``` for trigocherrin A), run a SPARQL query to check if your compound is present in Wikidata or not:


```SQL
SELECT ?item ?itemLabel WHERE {
  VALUES ?classes {
    wd:Q11173 # chemical compound
    wd:Q59199015 # group of stereoisomers
    wd:Q79529 # chemical substance
    wd:Q17339814 # group of chemical substances
    wd:Q47154513 # structural class of chemical compounds
  }
  ?item wdt:P31 ?classes. # instance of
  ?item wdt:P235 'QOVGHDRCAGYGEB-FFZYJECLSA-N'
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
}
```
[Try this query](https://w.wiki/32zU).
You can adapt it by replacing the InChIKey with the one for your compound.

Alternatively you can use the following Scholia link (replace by your compounds InChIKey) [https://scholia.toolforge.org/inchikey/QOVGHDRCAGYGEB-FFZYJECLSA-N](https://scholia.toolforge.org/inchikey/QOVGHDRCAGYGEB-FFZYJECLSA-N) 

If your compound is already present on Wikidata, you can directly skip to the [Add the biological source information](#add_bio) section below.

##### Add your data manually to Wikidata <a name="add_your_data_manually"></a>

First, if you do not have a Wikidata account already, it is advisable that you create one via [https://www.wikidata.org/wiki/Special:CreateAccount](https://www.wikidata.org/wiki/Special:CreateAccount).
While an account is not strictly required for manual edits, having one will be useful if you want to contribute more than once, and it helps in getting your contributions recognized.
Note that Wikidata accounts are integrated with accounts across the Wikimedia ecosystem, so if you already have an account on, say, any Wikipedia or on Wikispecies, then you can use the same credentials on Wikidata.

If you are unfamiliar with how Wikidata works, you can start by reading the Wikidata introduction page [https://www.wikidata.org/wiki/Wikidata:Introduction](https://www.wikidata.org/wiki/Wikidata:Introduction) and have a look at the Wikidata Tours page [https://www.wikidata.org/wiki/Wikidata:Tours](https://www.wikidata.org/wiki/Wikidata:Tours).

Now that you are all set up, you can go to Wikidata's page for creating new items, [https://www.wikidata.org/wiki/Special:NewItem](https://www.wikidata.org/wiki/Special:NewItem):


![](images/tutorial/2021-02-25-14-18-03.png){width="100%"}

An empty page with a new Wikidata identifier is created 

![](images/tutorial/2021-02-25-14-18-50.png){width="100%"}

###### Add the chemical compound information <a name="add_chemical"></a>

Create a new statement for ```is an instance of```

![](images/tutorial/2021-02-25-14-20-08.png){width="100%"}

and select chemical compound (i.e. [Q11173](https://www.wikidata.org/wiki/Q11173)):

![](images/tutorial/2021-02-25-14-20-58.png){width="100%"}

Click ```publish``` to save your changes and make them public.

Since you created a new item about an instance of a chemical compound, the user interface will automatically propose to you a set of additional statements commonly found on items about chemical compounds.

![](images/tutorial/2021-02-25-14-22-44.png){width="100%"}

You can then go on and fill these in.

Here, we start with the InChIKey.
Note the little flag which will automatically tell you if you have some problems with the recently created statements.

![](images/tutorial/2021-02-25-14-24-34.png){width="100%"}

Here, Wikidata tells us that if we add an InChiKey, we will need to also add an InChI.
Logical, but good to have a reminder!

Let's go ahead and add the InChI string.

Likewise, the addition of an isomeric SMILES string will require us to add a Canonical SMILES.

Note that you might have to copy and paste the SMILES string from PubChem to a plain text editor and then back to Wikidata because of some formatting issues when copy pasting directly from PubChem.

```BASH
C[C@@H]1C2[C@]([C@H]3[C@H]4[C@]1(C5=CC(=C(Cl)Cl)[C@@H]([C@]5([C@@H]([C@@]6([C@H]4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O3)(O2)C8=CC=CC=C8)(C(=C)C)O

CC1C2C(C3C4C1(C5=CC(=C(Cl)Cl)C(C5(C(C6(C4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O2)(O3)C8=CC=CC=C8)(C(=C)C)O
```

###### Add the biological source information <a name="add_bio"></a>

Now let's add the ```found in taxon``` property ([P703](https://www.wikidata.org/wiki/Property:P703)).

Just click on ```Add a new statement``` and type in the first letters of the property you want to add:

![](images/tutorial/2021-02-25-14-31-56.png){width="100%"}

Again, type in the first letters of the taxon, and if the organism is present, it will autocomplete.
Here is how this looks like for _Trigonostemon cherrieri_:

![](images/tutorial/2021-02-25-14-33-14.png){width="100%"}

Click ```publish``` to save your changes and make them public.


If your target taxon is not yet present on Wikidata and you are sure you have a valid taxon name that is spelled correctly, then you can go to [https://www.wikidata.org/wiki/Special:NewItem](https://www.wikidata.org/wiki/Special:NewItem), as described in the [Add your data manually to Wikidata](#add_your_data_manually) section.
For items about taxa, the ```instance of``` statement should have a value ```taxon``` (i.e. [Q16521](https://www.wikidata.org/wiki/Q16521)).
As for chemical compounds, the user interface will then suggest to you further statements to add.
For taxa, these include taxon name, parent taxon and taxon rank.


###### Add the reference documenting the structure-organism pair <a name="add_ref"></a>

Finally, since we report referenced structure-organisms pairs, we need to add the reference for this newly created ```compound found in taxon``` relationship.
This happens on the item about the compound, just below the ```found in taxon``` statement.
Click on the ```0 references``` link and then on ```add reference```:

![](images/tutorial/2021-02-25-14-35-33.png){width="100%"}

Here, we use the ```stated in``` property ([P248](https://www.wikidata.org/wiki/Property:P248)):

![](images/tutorial/2021-02-25-14-36-33.png){width="100%"}

Now, type in the first letters or word of the scientific publication documenting the natural product occurrence, autocompletion happens again.
Note that multiple publications might have the same title, and that there could be minor differences in punctuation or special characters between the information you and Wikidata have about the same reference.
If you are not sure whether your target reference is already in Wikidata, you can use its DOI to check, as outlined in the [Check whether your target reference is already on Wikidata](#check_ref) section.

![](images/tutorial/2021-02-25-14-38-54.png){width="100%"}

Click ```publish``` to save your changes and make them public.

![](images/tutorial/2021-02-25-14-39-48.png){width="100%"}

###### Check whether your target reference is already on Wikidata <a name="check_ref"></a>

If you are not sure whether your target reference is already in Wikidata, you can use its DOI to check.
For our DOI ```10.1021/ol2030907```, the URL [https://scholia.toolforge.org/doi/___10.1021/ol2030907___](https://scholia.toolforge.org/doi/10.1021/ol2030907) will lead you to a [Scholia](https://www.wikidata.org/wiki/Wikidata:Scholia) page about this publication: [https://scholia.toolforge.org/work/Q83059010](https://scholia.toolforge.org/work/Q83059010).
Scholia visualizes information from Wikidata, so if it has an entry for your target reference, then so does Wikidata, and both of them will use the same identifier (in this case [Q83059010](https://www.wikidata.org/wiki/Q83059010)).
If you prefer to resolve your DOI to Wikidata directly, you can do so by using the uppercase-normalized DOI in the following URL pattern: [https://hub.toolforge.org/P356:**10.1021/OL2030907**](https://hub.toolforge.org/P356:10.1021/OL2030907), which will lead you to the respective Wikidata page, in this case [Q83059010](https://www.wikidata.org/wiki/Q83059010).

If you think that no Wikidata entry exists for your target reference, you can use the DOI in the URL pattern [https://sourcemd.toolforge.org/index_old.php?id=**10.1021/ol2030907**&doit=Check+source](https://sourcemd.toolforge.org/index_old.php?id=10.1021/ol2030907&doit=Check+source), which will trigger a check with both Crossref and Wikidata, and if no Wikidata entry can be found, the metadata from Crossref will be fetched and presented to you for creating the respective Wikidata item semi-automatically.
Using such semi-automated workflows does require and account that is a minimum number of days old and has made a minimum number of edits on Wikidata.

If you are interested in the annotation of articles with topics via Wikidata and Scholia, here is a tutorial: [https://laurendupuis.github.io/Scholia_tutorial](https://laurendupuis.github.io/Scholia_tutorial).


##### Voilà !

You have added your first referenced structure-organism relationship to Wikidata and made a valuable contribution to the community.
You can add further statements, e.g. ```molecular formula```, or ```SPLASH code``` for linking to spectral data.

The Wikidata entry [https://www.wikidata.org/wiki/Q105674316](https://www.wikidata.org/wiki/Q105674316) has been started using these instructions.

You can run a SPARQL query and check that everything went smoothly by modifying the InChiKey line in the following [SPARQL query](https://w.wiki/32zb):

```SQL
SELECT ?item ?itemLabel ?taxonLabel ?artLabel WHERE {
  VALUES ?classes {
    wd:Q11173 # chemical compound
    wd:Q59199015 # group of stereoisomers
    wd:Q79529 # chemical substance
    wd:Q17339814 # group of chemical substances
    wd:Q47154513 # structural class of chemical compounds
  }
  ?item wdt:P31 ?classes. # instance of
  ?item wdt:P235 'QOVGHDRCAGYGEB-FFZYJECLSA-N' # InChiKey
  {
    ?item p:P1582 ?stmt. # natural product of taxon
    ?stmt ps:P1582 ?taxon. # natural product of taxon
    OPTIONAL {
      ?stmt prov:wasDerivedFrom ?ref.
      ?ref pr:P248 ?art. # stated in
    }
  }
  UNION
  {
    ?item p:P703 ?stmt. # found in taxon
    ?stmt ps:P703 ?taxon. # found in taxon
    OPTIONAL {
      ?stmt prov:wasDerivedFrom ?ref.
      ?ref pr:P248 ?art. # stated in
    }
  }
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
}
```


## References {.page_break_before}

<!-- Explicitly insert bibliography here -->
<div id="refs"></div>
